// @pandium/hubspot-client/dist/hubspot-client.js
import * as hubspot from '@hubspot/api-client';
var HubspotClient = /** @class */ (function () {
    function HubspotClient(abortController, baseUrl, apiToken, apiKey) {
        this.errorCount = 0;
        this.baseUrl = baseUrl;
        this.abortController = abortController;
        this.baseUrl = baseUrl;
        this.abortController = abortController;
        /*
        this.axiosInstance = axios.create({
            baseURL: this.baseUrl,
            timeout: 10000,
            headers: headers,
        })
  
  
        this.axiosInstance.interceptors.response.use(
            (resp: any) => {
                // reset ticker
                this.errorCount = 0
                return resp
            },
            async (error: AxiosError) => {
                this.errorCount++
                if (axios.isAxiosError(error)) {
                    // Kill the sync if we have an 401
                    if (error.response?.status === 401) {
                        console.error('Auth error. Stopping execution')
                        process.exit(1) // eslint-disable-line no-process-exit
                    }
                    if (this.errorCount > 10) {
                        console.error(
                            'Potential runaway error. Stopping execution'
                        )
                        process.exit(5) // eslint-disable-line no-process-exit
                    }
                }
                return Promise.reject(error)
            }
        )
         */
        var DEFAULT_LIMITER_OPTIONS = {
            minTime: 1000 / 9,
            maxConcurrent: 6,
            id: 'hubspot-client-limiter',
        };
        this.api = new hubspot.Client({
            accessToken: apiToken !== null && apiToken !== void 0 ? apiToken : process.env.PAN_SEC_HUBSPOT_OAUTH_ACCESS_TOKEN,
            developerApiKey: apiKey !== null && apiKey !== void 0 ? apiKey : process.env.PAN_SEC_HUBSPOT_BASIC_API_KEY,
            numberOfApiCallRetries: 3,
            limiterOptions: DEFAULT_LIMITER_OPTIONS,
        });
    }
    return HubspotClient;
}());
export { HubspotClient };
