// @hubspot/api-client/lib/codegen/cms/hubdb/types/ObjectParamAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectTablesApi = exports.ObjectRowsBatchApi = exports.ObjectRowsApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class ObjectRowsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableRowsApi(configuration, requestFactory, responseProcessor);
    }
    cloneDraftTableRow(param, options) {
        return this.api.cloneDraftTableRow(param.tableIdOrName, param.rowId, options).toPromise();
    }
    createTableRow(param, options) {
        return this.api.createTableRow(param.tableIdOrName, param.hubDbTableRowV3Request, options).toPromise();
    }
    getDraftTableRowById(param, options) {
        return this.api.getDraftTableRowById(param.tableIdOrName, param.rowId, options).toPromise();
    }
    getTableRow(param, options) {
        return this.api.getTableRow(param.tableIdOrName, param.rowId, options).toPromise();
    }
    getTableRows(param, options) {
        return this.api.getTableRows(param.tableIdOrName, param.sort, param.after, param.limit, param.properties, options).toPromise();
    }
    purgeDraftTableRow(param, options) {
        return this.api.purgeDraftTableRow(param.tableIdOrName, param.rowId, options).toPromise();
    }
    readDraftTableRows(param, options) {
        return this.api.readDraftTableRows(param.tableIdOrName, param.sort, param.after, param.limit, param.properties, options).toPromise();
    }
    replaceDraftTableRow(param, options) {
        return this.api.replaceDraftTableRow(param.tableIdOrName, param.rowId, param.hubDbTableRowV3Request, options).toPromise();
    }
    updateDraftTableRow(param, options) {
        return this.api.updateDraftTableRow(param.tableIdOrName, param.rowId, param.hubDbTableRowV3Request, options).toPromise();
    }
}
exports.ObjectRowsApi = ObjectRowsApi;
const ObservableAPI_2 = require("./ObservableAPI");
class ObjectRowsBatchApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableRowsBatchApi(configuration, requestFactory, responseProcessor);
    }
    cloneDraftTableRows(param, options) {
        return this.api.cloneDraftTableRows(param.tableIdOrName, param.batchInputString, options).toPromise();
    }
    createDraftTableRows(param, options) {
        return this.api.createDraftTableRows(param.tableIdOrName, param.batchInputHubDbTableRowV3Request, options).toPromise();
    }
    purgeDraftTableRows(param, options) {
        return this.api.purgeDraftTableRows(param.tableIdOrName, param.batchInputString, options).toPromise();
    }
    readDraftTableRows(param, options) {
        return this.api.readDraftTableRows(param.tableIdOrName, param.batchInputString, options).toPromise();
    }
    readTableRows(param, options) {
        return this.api.readTableRows(param.tableIdOrName, param.batchInputString, options).toPromise();
    }
    replaceDraftTableRows(param, options) {
        return this.api.replaceDraftTableRows(param.tableIdOrName, param.batchInputHubDbTableRowV3BatchUpdateRequest, options).toPromise();
    }
    updateDraftTableRows(param, options) {
        return this.api.updateDraftTableRows(param.tableIdOrName, param.batchInputHubDbTableRowV3BatchUpdateRequest, options).toPromise();
    }
}
exports.ObjectRowsBatchApi = ObjectRowsBatchApi;
const ObservableAPI_3 = require("./ObservableAPI");
class ObjectTablesApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_3.ObservableTablesApi(configuration, requestFactory, responseProcessor);
    }
    archiveTable(param, options) {
        return this.api.archiveTable(param.tableIdOrName, options).toPromise();
    }
    cloneDraftTable(param, options) {
        return this.api.cloneDraftTable(param.tableIdOrName, param.hubDbTableCloneRequest, options).toPromise();
    }
    createTable(param, options) {
        return this.api.createTable(param.hubDbTableV3Request, options).toPromise();
    }
    exportDraftTable(param, options) {
        return this.api.exportDraftTable(param.tableIdOrName, param.format, options).toPromise();
    }
    exportTable(param, options) {
        return this.api.exportTable(param.tableIdOrName, param.format, options).toPromise();
    }
    getAllDraftTables(param = {}, options) {
        return this.api.getAllDraftTables(param.sort, param.after, param.limit, param.createdAt, param.createdAfter, param.createdBefore, param.updatedAt, param.updatedAfter, param.updatedBefore, param.archived, options).toPromise();
    }
    getAllTables(param = {}, options) {
        return this.api.getAllTables(param.sort, param.after, param.limit, param.createdAt, param.createdAfter, param.createdBefore, param.updatedAt, param.updatedAfter, param.updatedBefore, param.archived, options).toPromise();
    }
    getDraftTableDetailsById(param, options) {
        return this.api.getDraftTableDetailsById(param.tableIdOrName, param.includeForeignIds, param.archived, options).toPromise();
    }
    getTableDetails(param, options) {
        return this.api.getTableDetails(param.tableIdOrName, param.includeForeignIds, param.archived, options).toPromise();
    }
    importDraftTable(param, options) {
        return this.api.importDraftTable(param.tableIdOrName, param.config, param.file, options).toPromise();
    }
    publishDraftTable(param, options) {
        return this.api.publishDraftTable(param.tableIdOrName, param.includeForeignIds, options).toPromise();
    }
    resetDraftTable(param, options) {
        return this.api.resetDraftTable(param.tableIdOrName, param.includeForeignIds, options).toPromise();
    }
    unpublishTable(param, options) {
        return this.api.unpublishTable(param.tableIdOrName, param.includeForeignIds, options).toPromise();
    }
    updateDraftTable(param, options) {
        return this.api.updateDraftTable(param.tableIdOrName, param.hubDbTableV3Request, param.includeForeignIds, param.archived, options).toPromise();
    }
}
exports.ObjectTablesApi = ObjectTablesApi;
