// @hubspot/api-client/lib/codegen/cms/hubdb/types/PromiseAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromiseTablesApi = exports.PromiseRowsBatchApi = exports.PromiseRowsApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class PromiseRowsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableRowsApi(configuration, requestFactory, responseProcessor);
    }
    cloneDraftTableRow(tableIdOrName, rowId, _options) {
        const result = this.api.cloneDraftTableRow(tableIdOrName, rowId, _options);
        return result.toPromise();
    }
    createTableRow(tableIdOrName, hubDbTableRowV3Request, _options) {
        const result = this.api.createTableRow(tableIdOrName, hubDbTableRowV3Request, _options);
        return result.toPromise();
    }
    getDraftTableRowById(tableIdOrName, rowId, _options) {
        const result = this.api.getDraftTableRowById(tableIdOrName, rowId, _options);
        return result.toPromise();
    }
    getTableRow(tableIdOrName, rowId, _options) {
        const result = this.api.getTableRow(tableIdOrName, rowId, _options);
        return result.toPromise();
    }
    getTableRows(tableIdOrName, sort, after, limit, properties, _options) {
        const result = this.api.getTableRows(tableIdOrName, sort, after, limit, properties, _options);
        return result.toPromise();
    }
    purgeDraftTableRow(tableIdOrName, rowId, _options) {
        const result = this.api.purgeDraftTableRow(tableIdOrName, rowId, _options);
        return result.toPromise();
    }
    readDraftTableRows(tableIdOrName, sort, after, limit, properties, _options) {
        const result = this.api.readDraftTableRows(tableIdOrName, sort, after, limit, properties, _options);
        return result.toPromise();
    }
    replaceDraftTableRow(tableIdOrName, rowId, hubDbTableRowV3Request, _options) {
        const result = this.api.replaceDraftTableRow(tableIdOrName, rowId, hubDbTableRowV3Request, _options);
        return result.toPromise();
    }
    updateDraftTableRow(tableIdOrName, rowId, hubDbTableRowV3Request, _options) {
        const result = this.api.updateDraftTableRow(tableIdOrName, rowId, hubDbTableRowV3Request, _options);
        return result.toPromise();
    }
}
exports.PromiseRowsApi = PromiseRowsApi;
const ObservableAPI_2 = require("./ObservableAPI");
class PromiseRowsBatchApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableRowsBatchApi(configuration, requestFactory, responseProcessor);
    }
    cloneDraftTableRows(tableIdOrName, batchInputString, _options) {
        const result = this.api.cloneDraftTableRows(tableIdOrName, batchInputString, _options);
        return result.toPromise();
    }
    createDraftTableRows(tableIdOrName, batchInputHubDbTableRowV3Request, _options) {
        const result = this.api.createDraftTableRows(tableIdOrName, batchInputHubDbTableRowV3Request, _options);
        return result.toPromise();
    }
    purgeDraftTableRows(tableIdOrName, batchInputString, _options) {
        const result = this.api.purgeDraftTableRows(tableIdOrName, batchInputString, _options);
        return result.toPromise();
    }
    readDraftTableRows(tableIdOrName, batchInputString, _options) {
        const result = this.api.readDraftTableRows(tableIdOrName, batchInputString, _options);
        return result.toPromise();
    }
    readTableRows(tableIdOrName, batchInputString, _options) {
        const result = this.api.readTableRows(tableIdOrName, batchInputString, _options);
        return result.toPromise();
    }
    replaceDraftTableRows(tableIdOrName, batchInputHubDbTableRowV3BatchUpdateRequest, _options) {
        const result = this.api.replaceDraftTableRows(tableIdOrName, batchInputHubDbTableRowV3BatchUpdateRequest, _options);
        return result.toPromise();
    }
    updateDraftTableRows(tableIdOrName, batchInputHubDbTableRowV3BatchUpdateRequest, _options) {
        const result = this.api.updateDraftTableRows(tableIdOrName, batchInputHubDbTableRowV3BatchUpdateRequest, _options);
        return result.toPromise();
    }
}
exports.PromiseRowsBatchApi = PromiseRowsBatchApi;
const ObservableAPI_3 = require("./ObservableAPI");
class PromiseTablesApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_3.ObservableTablesApi(configuration, requestFactory, responseProcessor);
    }
    archiveTable(tableIdOrName, _options) {
        const result = this.api.archiveTable(tableIdOrName, _options);
        return result.toPromise();
    }
    cloneDraftTable(tableIdOrName, hubDbTableCloneRequest, _options) {
        const result = this.api.cloneDraftTable(tableIdOrName, hubDbTableCloneRequest, _options);
        return result.toPromise();
    }
    createTable(hubDbTableV3Request, _options) {
        const result = this.api.createTable(hubDbTableV3Request, _options);
        return result.toPromise();
    }
    exportDraftTable(tableIdOrName, format, _options) {
        const result = this.api.exportDraftTable(tableIdOrName, format, _options);
        return result.toPromise();
    }
    exportTable(tableIdOrName, format, _options) {
        const result = this.api.exportTable(tableIdOrName, format, _options);
        return result.toPromise();
    }
    getAllDraftTables(sort, after, limit, createdAt, createdAfter, createdBefore, updatedAt, updatedAfter, updatedBefore, archived, _options) {
        const result = this.api.getAllDraftTables(sort, after, limit, createdAt, createdAfter, createdBefore, updatedAt, updatedAfter, updatedBefore, archived, _options);
        return result.toPromise();
    }
    getAllTables(sort, after, limit, createdAt, createdAfter, createdBefore, updatedAt, updatedAfter, updatedBefore, archived, _options) {
        const result = this.api.getAllTables(sort, after, limit, createdAt, createdAfter, createdBefore, updatedAt, updatedAfter, updatedBefore, archived, _options);
        return result.toPromise();
    }
    getDraftTableDetailsById(tableIdOrName, includeForeignIds, archived, _options) {
        const result = this.api.getDraftTableDetailsById(tableIdOrName, includeForeignIds, archived, _options);
        return result.toPromise();
    }
    getTableDetails(tableIdOrName, includeForeignIds, archived, _options) {
        const result = this.api.getTableDetails(tableIdOrName, includeForeignIds, archived, _options);
        return result.toPromise();
    }
    importDraftTable(tableIdOrName, config, file, _options) {
        const result = this.api.importDraftTable(tableIdOrName, config, file, _options);
        return result.toPromise();
    }
    publishDraftTable(tableIdOrName, includeForeignIds, _options) {
        const result = this.api.publishDraftTable(tableIdOrName, includeForeignIds, _options);
        return result.toPromise();
    }
    resetDraftTable(tableIdOrName, includeForeignIds, _options) {
        const result = this.api.resetDraftTable(tableIdOrName, includeForeignIds, _options);
        return result.toPromise();
    }
    unpublishTable(tableIdOrName, includeForeignIds, _options) {
        const result = this.api.unpublishTable(tableIdOrName, includeForeignIds, _options);
        return result.toPromise();
    }
    updateDraftTable(tableIdOrName, hubDbTableV3Request, includeForeignIds, archived, _options) {
        const result = this.api.updateDraftTable(tableIdOrName, hubDbTableV3Request, includeForeignIds, archived, _options);
        return result.toPromise();
    }
}
exports.PromiseTablesApi = PromiseTablesApi;
