// @pandium/apollo-io-client/dist/apollo-io.js
import { makeApi, Zodios } from "@zodios/core";
import { peopleEnrichmentMutationResponseSchema, peopleEnrichmentQueryParamsSchema, peopleEnrichment401Schema, peopleEnrichment429Schema } from "./schemas/peopleEnrichmentSchema.js";
import { bulkPeopleEnrichmentMutationResponseSchema, bulkPeopleEnrichmentQueryParamsSchema, bulkPeopleEnrichmentMutationRequestSchema, bulkPeopleEnrichment400Schema, bulkPeopleEnrichment401Schema, bulkPeopleEnrichment422Schema, bulkPeopleEnrichment429Schema } from "./schemas/bulkPeopleEnrichmentSchema.js";
import { organizationEnrichmentQueryResponseSchema, organizationEnrichmentQueryParamsSchema, organizationEnrichment401Schema, organizationEnrichment422Schema, organizationEnrichment429Schema } from "./schemas/organizationEnrichmentSchema.js";
import { bulkOrganizationEnrichmentMutationResponseSchema, bulkOrganizationEnrichmentQueryParamsSchema, bulkOrganizationEnrichment400Schema, bulkOrganizationEnrichment401Schema, bulkOrganizationEnrichment429Schema } from "./schemas/bulkOrganizationEnrichmentSchema.js";
import { peopleSearchMutationResponseSchema, peopleSearchQueryParamsSchema, peopleSearch401Schema, peopleSearch403Schema, peopleSearch429Schema } from "./schemas/peopleSearchSchema.js";
import { organizationSearchMutationResponseSchema, organizationSearchQueryParamsSchema, organizationSearch401Schema, organizationSearch403Schema, organizationSearch429Schema } from "./schemas/organizationSearchSchema.js";
import { organizationJobsPostingsQueryResponseSchema, organizationJobsPostingsPathParamsSchema, organizationJobsPostings401Schema, organizationJobsPostings422Schema, organizationJobsPostings429Schema } from "./schemas/organizationJobsPostingsSchema.js";
import { createDealMutationResponseSchema, createDealQueryParamsSchema, createDeal401Schema, createDeal403Schema, createDeal422Schema, createDeal429Schema } from "./schemas/createDealSchema.js";
import { listAllDealsQueryResponseSchema, listAllDealsQueryParamsSchema, listAllDeals401Schema, listAllDeals403Schema, listAllDeals429Schema } from "./schemas/listAllDealsSchema.js";
import { viewDealQueryResponseSchema, viewDealPathParamsSchema, viewDeal401Schema, viewDeal403Schema, viewDeal404Schema, viewDeal422Schema, viewDeal429Schema } from "./schemas/viewDealSchema.js";
import { updateDealMutationResponseSchema, updateDealPathParamsSchema, updateDealQueryParamsSchema, updateDeal401Schema, updateDeal403Schema, updateDeal422Schema, updateDeal429Schema } from "./schemas/updateDealSchema.js";
import { listDealStagesQueryResponseSchema, listDealStages401Schema, listDealStages403Schema, listDealStages429Schema } from "./schemas/listDealStagesSchema.js";
import { createAContactMutationResponseSchema, createAContactQueryParamsSchema, createAContact401Schema, createAContact422Schema, createAContact429Schema } from "./schemas/createAContactSchema.js";
import { updateAContactMutationResponseSchema, updateAContactPathParamsSchema, updateAContactQueryParamsSchema, updateAContact401Schema, updateAContact422Schema, updateAContact429Schema } from "./schemas/updateAContactSchema.js";
import { searchForContactsMutationResponseSchema, searchForContactsQueryParamsSchema, searchForContacts401Schema, searchForContacts403Schema, searchForContacts429Schema } from "./schemas/searchForContactsSchema.js";
import { listContactStagesQueryResponseSchema, listContactStages401Schema, listContactStages403Schema, listContactStages429Schema } from "./schemas/listContactStagesSchema.js";
import { updateContactStageMutationResponseSchema, updateContactStageQueryParamsSchema, updateContactStage401Schema, updateContactStage422Schema, updateContactStage429Schema } from "./schemas/updateContactStageSchema.js";
import { updateContactOwnershipMutationResponseSchema, updateContactOwnershipQueryParamsSchema, updateContactOwnership401Schema, updateContactOwnership422Schema, updateContactOwnership429Schema } from "./schemas/updateContactOwnershipSchema.js";
import { createAnAccountMutationResponseSchema, createAnAccountQueryParamsSchema, createAnAccount401Schema, createAnAccount403Schema, createAnAccount422Schema, createAnAccount429Schema } from "./schemas/createAnAccountSchema.js";
import { updateAnAccountMutationResponseSchema, updateAnAccountPathParamsSchema, updateAnAccountQueryParamsSchema, updateAnAccount401Schema, updateAnAccount403Schema, updateAnAccount422Schema, updateAnAccount429Schema } from "./schemas/updateAnAccountSchema.js";
import { searchForAccountsMutationResponseSchema, searchForAccountsQueryParamsSchema, searchForAccounts401Schema, searchForAccounts403Schema, searchForAccounts422Schema, searchForAccounts429Schema } from "./schemas/searchForAccountsSchema.js";
import { listAccountStagesQueryResponseSchema, listAccountStages401Schema, listAccountStages403Schema, listAccountStages429Schema } from "./schemas/listAccountStagesSchema.js";
import { updateAccountStageMutationResponseSchema, updateAccountStageQueryParamsSchema, updateAccountStage401Schema, updateAccountStage403Schema, updateAccountStage429Schema } from "./schemas/updateAccountStageSchema.js";
import { updateAccountOwnershipMutationResponseSchema, updateAccountOwnershipQueryParamsSchema, updateAccountOwnership401Schema, updateAccountOwnership403Schema, updateAccountOwnership422Schema, updateAccountOwnership429Schema } from "./schemas/updateAccountOwnershipSchema.js";
import { searchForSequencesMutationResponseSchema, searchForSequencesQueryParamsSchema, searchForSequences401Schema, searchForSequences403Schema, searchForSequences429Schema } from "./schemas/searchForSequencesSchema.js";
import { addContactsToSequenceMutationResponseSchema, addContactsToSequencePathParamsSchema, addContactsToSequenceQueryParamsSchema, addContactsToSequence401Schema, addContactsToSequence403Schema, addContactsToSequence422Schema, addContactsToSequence429Schema } from "./schemas/addContactsToSequenceSchema.js";
import { updateContactStatusSequenceMutationResponseSchema, updateContactStatusSequenceQueryParamsSchema, updateContactStatusSequence401Schema, updateContactStatusSequence403Schema, updateContactStatusSequence429Schema } from "./schemas/updateContactStatusSequenceSchema.js";
import { createTaskMutationResponseSchema, createTaskQueryParamsSchema, createTask401Schema, createTask403Schema, createTask429Schema } from "./schemas/createTaskSchema.js";
import { searchTasksMutationResponseSchema, searchTasksQueryParamsSchema, searchTasks401Schema, searchTasks403Schema, searchTasks422Schema, searchTasks429Schema } from "./schemas/searchTasksSchema.js";
import { getAListOfUsersQueryResponseSchema, getAListOfUsersQueryParamsSchema, getAListOfUsers401Schema, getAListOfUsers403Schema, getAListOfUsers429Schema } from "./schemas/getAListOfUsersSchema.js";
import { getAListOfEmailAccountsQueryResponseSchema, getAListOfEmailAccounts401Schema, getAListOfEmailAccounts403Schema, getAListOfEmailAccounts429Schema } from "./schemas/getAListOfEmailAccountsSchema.js";
import { getAListOfAllListstagsQueryResponseSchema, getAListOfAllListstags401Schema, getAListOfAllListstags403Schema, getAListOfAllListstags429Schema } from "./schemas/getAListOfAllListstagsSchema.js";
import { getAListOfAllCustomFieldsQueryResponseSchema, getAListOfAllCustomFields401Schema, getAListOfAllCustomFields403Schema, getAListOfAllCustomFields429Schema } from "./schemas/getAListOfAllCustomFieldsSchema.js";
export var endpoints = makeApi([
    {
        method: "post",
        path: "/people/match",
        description: "Use the People Enrichment endpoint to enrich data for 1 person. To enrich data for up to 10 people with a single API call, use the <a href=\"https://docs.apollo.io/reference/bulk-people-enrichment\" target=\"_blank\">Bulk People Enrichment endpoint</a> instead.  <br><br>Apollo relies on the information you pass via the endpoint's parameters to identify the correct person to enrich. If you provide more information about a person, Apollo is more likely to find a match within its database. If you only provide general information, such as a name without a domain or email address, you might receive a 200 response, but the response will indicate that no records have been enriched. <br><br>By default, this endpoint does not return personal emails or phone numbers. Use the `reveal_personal_emails` and `reveal_phone_number` parameters to retrieve emails and phone numbers. These options do potentially consume credits as part of your <a href=\"https://docs.apollo.io/docs/api-pricing\" target=\"_blank\">Apollo pricing plan</a>.",
        requestFormat: "json",
        parameters: [
            {
                name: "first_name",
                description: "The first name of the person. This is typically used in combination with the `last_name` parameter. <br><br>Example: `tim`",
                type: "Query",
                schema: peopleEnrichmentQueryParamsSchema.shape["first_name"]
            },
            {
                name: "last_name",
                description: "The last name of the person. This is typically used in combination with the `first_name` parameter. <br><br>Example: `zheng`",
                type: "Query",
                schema: peopleEnrichmentQueryParamsSchema.shape["last_name"]
            },
            {
                name: "name",
                description: "The full name of the person. This will typically be a first name and last name separated by a space.  If you use this parameter, you do not need to use the `first_name` and `last_name` parameters. <br><br>Example: `tim zheng`",
                type: "Query",
                schema: peopleEnrichmentQueryParamsSchema.shape["name"]
            },
            {
                name: "email",
                description: "The email address of the person. <br><br>Example: `example@email.com`",
                type: "Query",
                schema: peopleEnrichmentQueryParamsSchema.shape["email"]
            },
            {
                name: "hashed_email",
                description: "The hashed email of the person. The email should adhere to either the MD5 or SHA-256 hash format. <br><br>Example: `8d935115b9ff4489f2d1f9249503cadf` (MD5) or `97817c0c49994eb500ad0a5e7e2d8aed51977b26424d508f66e4e8887746a152` (SHA-256)",
                type: "Query",
                schema: peopleEnrichmentQueryParamsSchema.shape["hashed_email"]
            },
            {
                name: "organization_name",
                description: "The name of the person's employer. This can be the current employer or a previous employer. <br><br>Example: `apollo`",
                type: "Query",
                schema: peopleEnrichmentQueryParamsSchema.shape["organization_name"]
            },
            {
                name: "domain",
                description: "The domain name for the person's employer. This can be the current employer or a previous employer. Do not include `www.`, the `@` symbol, or similar. <br><br>Example: `apollo.io` or `microsoft.com`",
                type: "Query",
                schema: peopleEnrichmentQueryParamsSchema.shape["domain"]
            },
            {
                name: "id",
                description: "The Apollo ID for the person. Each person in the Apollo database is assigned a unique ID. <br><br>To find IDs, call the <a href=\"https://docs.apollo.io/reference/people-search\" target=\"_blank\">People Search endpoint</a> and identify the values for `person_id`. <br><br>Example: `587cf802f65125cad923a266`",
                type: "Query",
                schema: peopleEnrichmentQueryParamsSchema.shape["id"]
            },
            {
                name: "linkedin_url",
                description: "The URL for the person's LinkedIn profile. <br><br>Example: `http://www.linkedin.com/in/tim-zheng-677ba010`",
                type: "Query",
                schema: peopleEnrichmentQueryParamsSchema.shape["linkedin_url"]
            },
            {
                name: "reveal_personal_emails",
                description: "Set to `true` if you want to enrich the person's data with personal emails. This potentially consumes credits as part of your <a href=\"https://docs.apollo.io/docs/api-pricing\" target=\"_blank\">Apollo pricing plan</a>. The default value is `false`. <br><br>If a person resides in a <a href=\"https://knowledge.apollo.io/hc/en-us/articles/4409141087757\" target=\"_blank\">GDPR</a>-compliant region, Apollo will not reveal their personal email.",
                type: "Query",
                schema: peopleEnrichmentQueryParamsSchema.shape["reveal_personal_emails"]
            },
            {
                name: "reveal_phone_number",
                description: "Set to `true` if you want to enrich the person's data with all available phone numbers, including mobile phone numbers. This potentially consumes credits as part of your <a href=\"https://docs.apollo.io/docs/api-pricing\" target=\"_blank\">Apollo pricing plan</a>. The default value is `false`. <br><br>If this parameter is set to `true`, you must enter a webhook URL for the `webhook_url` parameter. Apollo will asynchronously verify phone numbers for you, then send a JSON response that includes only details about the person's phone numbers to the webhook URL you provide. It can take several minutes for the phone numbers to be delivered.",
                type: "Query",
                schema: peopleEnrichmentQueryParamsSchema.shape["reveal_phone_number"]
            },
            {
                name: "webhook_url",
                description: "If you set the `reveal_phone_number` parameter to `true`, this parameter becomes mandatory. Otherwise, do not use this parameter. <br><br>Enter the webhook URL that specifies where Apollo should send a JSON response that includes the phone number you requested. Apollo suggests testing this flow to ensure you receive the separate response with the phone number. <br><br>Example: `https://webhook.site/cc4cf44e-e047-4774-8dac-473d28474e40`",
                type: "Query",
                schema: peopleEnrichmentQueryParamsSchema.shape["webhook_url"]
            }
        ],
        response: peopleEnrichmentMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: peopleEnrichment401Schema
            },
            {
                status: 429,
                description: "429",
                schema: peopleEnrichment429Schema
            }
        ],
        alias: "peopleEnrichment",
    },
    {
        method: "post",
        path: "/people/bulk_match",
        description: "Use the Bulk People Enrichment endpoint to enrich data for up to 10 people with a single API call. To enrich data for only 1 person, use the <a href=\"https://docs.apollo.io/reference/people-enrichment\" target=\"_blank\">People Enrichment endpoint</a> instead. <br><br>Apollo relies on the information you pass via the endpoint's parameters to identify the correct people to enrich. When you provide more information, Apollo is more likely to find matches within its database. If you only provide general information, such as a name without a domain or email address, you might receive a `200` response, but the response will indicate that no records have been enriched. The details for each person should be passed as an object with the `details[]` array. <br><br>By default, this endpoint does not return personal emails or phone numbers. Use the `reveal_personal_emails` and `reveal_phone_number` parameters to retrieve emails and phone numbers. If you set either of these parameters to `true`, Apollo will attempt to provide emails or phone numbers for all matches. These options do potentially consume credits as part of your <a href=\"https://docs.apollo.io/docs/api-pricing\" target=\"_blank\">Apollo pricing plan</a>. <br><br>This endpoint's <a href=\"https://docs.apollo.io/reference/rate-limits\" target=\"_blank\">rate limit</a> is throttled to 10% of the People Enrichment endpoint.",
        requestFormat: "json",
        parameters: [
            {
                name: "reveal_personal_emails",
                description: "Set to `true` if you want to enrich all matched people with personal emails. This potentially consumes credits as part of your <a href=\"https://docs.apollo.io/docs/api-pricing\" target=\"_blank\">Apollo pricing plan</a>. The default value is `false`. <br><br>If a person resides in a <a href=\"https://knowledge.apollo.io/hc/en-us/articles/4409141087757\" target=\"_blank\">GDPR</a>-compliant region, Apollo will not reveal their personal email.",
                type: "Query",
                schema: bulkPeopleEnrichmentQueryParamsSchema.shape["reveal_personal_emails"]
            },
            {
                name: "reveal_phone_number",
                description: "Set to `true` if you want to enrich the data of all matched people with all available phone numbers, including mobile phone numbers. This potentially consumes credits as part of your <a href=\"https://docs.apollo.io/docs/api-pricing\" target=\"_blank\">Apollo pricing plan</a>. The default value is `false`. <br><br>If this parameter is set to `true`, you must enter a webhook URL for the `webhook_url` parameter. Apollo will asynchronously verify phone numbers for you, then send a JSON response that includes only details about the phone numbers to the webhook URL you provide. It can take several minutes for the phone numbers to be delivered.",
                type: "Query",
                schema: bulkPeopleEnrichmentQueryParamsSchema.shape["reveal_phone_number"]
            },
            {
                name: "webhook_url",
                description: "If you set the `reveal_phone_number` parameter to `true`, this parameter becomes mandatory. Otherwise, do not use this parameter. <br><br>Enter the webhook URL that specifies where Apollo should send a JSON response that includes the phone numbers you requested. Apollo suggests testing this flow to ensure you receive the separate response with the phone numbers. <br><br>Example: `https://webhook.site/cc4cf44e-e047-4774-8dac-473d28474e40`",
                type: "Query",
                schema: bulkPeopleEnrichmentQueryParamsSchema.shape["webhook_url"]
            },
            {
                name: "BulkPeopleEnrichmentMutationRequest",
                description: "",
                type: "Body",
                schema: bulkPeopleEnrichmentMutationRequestSchema
            }
        ],
        response: bulkPeopleEnrichmentMutationResponseSchema,
        errors: [
            {
                status: 400,
                description: "400",
                schema: bulkPeopleEnrichment400Schema
            },
            {
                status: 401,
                description: "401",
                schema: bulkPeopleEnrichment401Schema
            },
            {
                status: 422,
                description: "422",
                schema: bulkPeopleEnrichment422Schema
            },
            {
                status: 429,
                description: "429",
                schema: bulkPeopleEnrichment429Schema
            }
        ],
        alias: "bulkPeopleEnrichment",
    },
    {
        method: "get",
        path: "/organizations/enrich",
        description: "Use the Oganization Enrichment endpoint to enrich data for 1 company.  To enrich data for up to 10 companies with a single API call, use the <a href=\"https://docs.apollo.io/reference/bulk-organization-enrichment\" target=\"_blank\">Bulk Organization Enrichment endpoint</a> instead. <br><br>Enriched data potentially includes industry information, revenue, employee counts, funding round details, and corporate phone numbers and locations.",
        requestFormat: "json",
        parameters: [
            {
                name: "domain",
                description: "The domain of the company that you want to enrich. Do not include `www.`, the `@` symbol, or similar. <br><br>Example: `apollo.io` or `microsoft.com`",
                type: "Query",
                schema: organizationEnrichmentQueryParamsSchema.shape["domain"]
            }
        ],
        response: organizationEnrichmentQueryResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: organizationEnrichment401Schema
            },
            {
                status: 422,
                description: "422",
                schema: organizationEnrichment422Schema
            },
            {
                status: 429,
                description: "429",
                schema: organizationEnrichment429Schema
            }
        ],
        alias: "organizationEnrichment",
    },
    {
        method: "post",
        path: "/organizations/bulk_enrich",
        description: "Use the Bulk Organization Enrichment endpoint to enrich data for up to 10 companies with a single API call. To enrich data for only 1 company, use the <a href=\"https://docs.apollo.io/reference/organization-enrichment\" target=\"_blank\">Organization Enrichment endpoint</a> instead.<br><br>Enriched data potentially includes industry information, revenue, employee counts, funding round details, and corporate phone numbers and locations. <br><br>This endpoint's <a href=\"https://docs.apollo.io/reference/rate-limits\" target=\"_blank\">rate limit</a> is throttled to 10% of the Organization Enrichment endpoint.",
        requestFormat: "json",
        parameters: [
            {
                name: "domains[]",
                description: "The domain of each company that you want to enrich. Do not include `www.`, the `@` symbol, or similar. <br><br>Example: `apollo.io` and `microsoft.com`",
                type: "Query",
                schema: bulkOrganizationEnrichmentQueryParamsSchema.shape["domains[]"]
            }
        ],
        response: bulkOrganizationEnrichmentMutationResponseSchema,
        errors: [
            {
                status: 400,
                description: "400",
                schema: bulkOrganizationEnrichment400Schema
            },
            {
                status: 401,
                description: "401",
                schema: bulkOrganizationEnrichment401Schema
            },
            {
                status: 429,
                description: "429",
                schema: bulkOrganizationEnrichment429Schema
            }
        ],
        alias: "bulkOrganizationEnrichment",
    },
    {
        method: "post",
        path: "/mixed_people/search",
        description: "Use the People Search endpoint to find people in the Apollo database. Several filters are available to help narrow your search. <br><br>Calling this endpoint does consume credits as part of your <a href=\"https://docs.apollo.io/docs/api-pricing\" target=\"_blank\">Apollo pricing plan</a>. This feature is not accessible to Apollo users on free plans.<br><br>This endpoint does not return new email addresses or phone numbers. Use the <a href=\"https://docs.apollo.io/reference/people-enrichment\" target=\"_blank\">People Enrichment</a> or <a href=\"https://docs.apollo.io/reference/bulk-people-enrichment\" target=\"_blank\">Bulk People Enrichment</a> endpoints to enrich data. <br><br>To protect Apollo's performance for all users, this endpoint has a display limit of 50,000 records (100 records per page, up to 500 pages). Add more filters to narrow your search results as much as possible. This limitation does not restrict your access to Apollo's database; you just need to access the data in batches.",
        requestFormat: "json",
        parameters: [
            {
                name: "person_titles[]",
                description: "Job titles held by the people you want to find. For a person to be included in search results, they only need to match 1 of the job titles you add. Adding more job titles expands your search results. <br><br>Results also include job titles with the same terms, even if they are not exact matches. For example, searching for `marketing manager` might return people with the job title `content marketing manager`. <br><br>Use this parameter in combination with the `person_seniorities[]` parameter to find people based on specific job functions and seniority levels. <br><br>Examples: `sales development representative`; `marketing manager`; `research analyst`",
                type: "Query",
                schema: peopleSearchQueryParamsSchema.shape["person_titles[]"]
            },
            {
                name: "person_locations[]",
                description: "The location where people live. You can search across cities, US states, and countries. <br><br>To find people based on the headquarters locations of their current employer, use the `organization_locations` parameter. <br><br>Examples: `california`; `ireland`; `chicago`",
                type: "Query",
                schema: peopleSearchQueryParamsSchema.shape["person_locations[]"]
            },
            {
                name: "person_seniorities[]",
                description: "The job seniority that people hold within their current employer. This enables you to find people that currently hold positions at certain reporting levels, such as Director level or senior IC level. For a person to be included in search results, they only need to match 1 of the seniorities you add. Adding more seniorities expands your search results. <br><br>Searches only return results based on their current job title, so searching for Director-level employees only returns people that currently hold a Director-level title. If someone was previously a Director, but is currently a VP, they would not be included in your search results. <br><br>Use this parameter in combination with the `person_titles[]` parameter to find people based on specific job functions and seniority levels. <br><br>Example: `director`; `senior`; `vp`",
                type: "Query",
                schema: peopleSearchQueryParamsSchema.shape["person_seniorities[]"]
            },
            {
                name: "organization_locations[]",
                description: "The location of the company headquarters for a person's current employer. You can search across cities, US states, and countries. <br><br>If a company has several office locations, results are still based on the headquarters location. For example, if you search `chicago` but a company's HQ location is in `boston`, people that work for the Boston-based company will not appear in your results, even if they match other parameters. <br><br>To find people based on their personal location, use the `person_locations` parameter. <br><br>Examples: `texas`; `tokyo`; `spain`",
                type: "Query",
                schema: peopleSearchQueryParamsSchema.shape["organization_locations[]"]
            },
            {
                name: "q_organization_domains",
                description: "The domain name for the person's employer. This can be the current employer or a previous employer. Do not include `www.`, the `@` symbol, or similar. <br><br>You can add multiple domains to search across companies. <br><br>Examples: `apollo.io` and `microsoft.com`",
                type: "Query",
                schema: peopleSearchQueryParamsSchema.shape["q_organization_domains"]
            },
            {
                name: "contact_email_status[]",
                description: "The email statuses for the people you want to find. You can add multiple statuses to expand your search. <br><br>The statuses you can search include:  <ul> <li> <code>verified</code> </li> <li> <code>unverified</code> </li> <li> <code>likely to engage</code> </li> <li> <code>unavailable</code> </li>  </ul>",
                type: "Query",
                schema: peopleSearchQueryParamsSchema.shape["contact_email_status[]"]
            },
            {
                name: "organization_ids[]",
                description: "The Apollo IDs for the companies (employers) you want to include in your search results. Each company in the Apollo database is assigned a unique ID. <br><br>To find IDs, call the <a href=\"https://docs.apollo.io/reference/organization-search\" target=\"_blank\">Organization Search endpoint</a> and identify the values for `organization_id`.  <br><br>Example: `5e66b6381e05b4008c8331b8`",
                type: "Query",
                schema: peopleSearchQueryParamsSchema.shape["organization_ids[]"]
            },
            {
                name: "organization_num_employees_ranges[]",
                description: "The number range of employees working for the person's current company. This enables you to find people based on the headcount of their employer. You can add multiple ranges to expand your search results. <br><br>Each range you add needs to be a string, with the upper and lower numbers of the range separated only by a comma. <br><br>Examples: `1,10`; `250,500`; `10000,20000`",
                type: "Query",
                schema: peopleSearchQueryParamsSchema.shape["organization_num_employees_ranges[]"]
            },
            {
                name: "q_keywords",
                description: "A string of words over which we want to filter the results",
                type: "Query",
                schema: peopleSearchQueryParamsSchema.shape["q_keywords"]
            },
            {
                name: "page",
                description: "The page number of the Apollo data that you want to retrieve. <br><br>Use this parameter in combination with the `per_page` parameter to make search results for navigable and improve the performance of the endpoint. <br><br>Example: `4`",
                type: "Query",
                schema: peopleSearchQueryParamsSchema.shape["page"]
            },
            {
                name: "per_page",
                description: "\"The number of search results that should be returned for each page. Limited the number of results per page improves the endpoint's performance. <br><br>Use the `page` parameter to search the different pages of data. <br><br>Example: `10`",
                type: "Query",
                schema: peopleSearchQueryParamsSchema.shape["per_page"]
            }
        ],
        response: peopleSearchMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: peopleSearch401Schema
            },
            {
                status: 403,
                description: "403",
                schema: peopleSearch403Schema
            },
            {
                status: 429,
                description: "429",
                schema: peopleSearch429Schema
            }
        ],
        alias: "searchPeople",
    },
    {
        method: "post",
        path: "/mixed_companies/search",
        description: "Use the Organization Search endpoint to find companies in the Apollo database. Several filters are available to help narrow your search. <br><br>Calling this endpoint does consume credits as part of your <a href=\"https://docs.apollo.io/docs/api-pricing\" target=\"_blank\">Apollo pricing plan</a>. This feature is not accessible to Apollo users on free plans.<br><br>To protect Apollo's performance for all users, this endpoint has a display limit of 50,000 records (100 records per page, up to 500 pages). Add more filters to narrow your search results as much as possible. This limitation does not restrict your access to Apollo's database; you just need to access the data in batches.",
        requestFormat: "json",
        parameters: [
            {
                name: "organization_num_employees_ranges[]",
                description: "The number range of employees working for the company. This enables you to find companies based on headcount. You can add multiple ranges to expand your search results. <br><br>Each range you add needs to be a string, with the upper and lower numbers of the range separated only by a comma. <br><br>Examples: `1,10`; `250,500`; `10000,20000`",
                type: "Query",
                schema: organizationSearchQueryParamsSchema.shape["organization_num_employees_ranges[]"]
            },
            {
                name: "organization_locations[]",
                description: "The location of the company headquarters. You can search across cities, US states, and countries. <br><br>If a company has several office locations, results are still based on the headquarters location. For example, if you search `chicago` but a company's HQ location is in `boston`, any Boston-based companies will not appearch in your search results, even if they match other parameters. <br><br>To exclude companies based on location, use the `organization_not_locations` parameter. <br><br>Examples: `texas`; `tokyo`; `spain`",
                type: "Query",
                schema: organizationSearchQueryParamsSchema.shape["organization_locations[]"]
            },
            {
                name: "organization_not_locations[]",
                description: "Exclude companies from search results based on the location of the company headquarters. You can use cities, US states, and countries as locations to exclude. <br><br>This parameter is useful for ensuring you do not prospect in an undesirable territory. For example, if you use `ireland` as a value, no Ireland-based companies will appear in your search results. <br><br>Examples: `minnesota`; `ireland`; `seoul`",
                type: "Query",
                schema: organizationSearchQueryParamsSchema.shape["organization_not_locations[]"]
            },
            {
                name: "q_organization_keyword_tags[]",
                description: "Filter search results based on keywords associated with companies. For example, you can enter `mining` as a value to return only companies that have an association with the mining industry. <br><br>Examples: `mining`; `sales strategy`; `consulting`",
                type: "Query",
                schema: organizationSearchQueryParamsSchema.shape["q_organization_keyword_tags[]"]
            },
            {
                name: "q_organization_name",
                description: "Filter search results to include a specific company name. <br><br>If the value you enter for this parameter does not match with a company's name, the company will not appear in search results, even if it matches other parameters. Partial matches are accepted. For example, if you filter by the value `marketing`, a company called `NY Marketing Unlimited` would still be eligible as a search result, but `NY Market Analysis` would not be eligible. <br><br>Example: `apollo` or `mining`",
                type: "Query",
                schema: organizationSearchQueryParamsSchema.shape["q_organization_name"]
            },
            {
                name: "organization_ids[]",
                description: "The Apollo IDs for the companies you want to include in your search results. Each company in the Apollo database is assigned a unique ID. <br><br>To find IDs, identify the values for `organization_id` when you call this endpoint. <br><br>Example: `5e66b6381e05b4008c8331b8`",
                type: "Query",
                schema: organizationSearchQueryParamsSchema.shape["organization_ids[]"]
            },
            {
                name: "page",
                description: "The page number of the Apollo data that you want to retrieve. <br><br>Use this parameter in combination with the `per_page` parameter to make search results for navigable and improve the performance of the endpoint. <br><br>Example: `4`",
                type: "Query",
                schema: organizationSearchQueryParamsSchema.shape["page"]
            },
            {
                name: "per_page",
                description: "The number of search results that should be returned for each page. Limited the number of results per page improves the endpoint's performance. <br><br>Use the `page` parameter to search the different pages of data. <br><br>Example: `10`",
                type: "Query",
                schema: organizationSearchQueryParamsSchema.shape["per_page"]
            }
        ],
        response: organizationSearchMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: organizationSearch401Schema
            },
            {
                status: 403,
                description: "403",
                schema: organizationSearch403Schema
            },
            {
                status: 429,
                description: "429",
                schema: organizationSearch429Schema
            }
        ],
        alias: "searchOrganizations",
    },
    {
        method: "get",
        path: "/organizations/:organization_id/job_postings",
        description: "Use the Organization Job Postings endpoint to retrieve the current job postings for companies. This can help you identify companies that are growing headcount in areas that are strategically important for you.",
        requestFormat: "json",
        parameters: [
            {
                name: "organization_id",
                description: "The organization ID of the company for which you want to find job postings. <br><br>Each company in the Apollo database is assigned a unique ID. To find IDs, call the <a href=\"https://docs.apollo.io/reference/organization-search\" target=\"_blank\">Organization Search endpoint</a> and identify the values for `organization_id`. <br><br>Example: `5e66b6381e05b4008c8331b8`",
                type: "Path",
                schema: organizationJobsPostingsPathParamsSchema.shape["organization_id"]
            }
        ],
        response: organizationJobsPostingsQueryResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: organizationJobsPostings401Schema
            },
            {
                status: 422,
                description: "422",
                schema: organizationJobsPostings422Schema
            },
            {
                status: 429,
                description: "429",
                schema: organizationJobsPostings429Schema
            }
        ],
        alias: "listOrganizationJobPostings",
    },
    {
        method: "post",
        path: "/opportunities",
        description: "Use the Create Deal endpoint to create new deals for an Apollo account. Deals enable you to track account activity, including monetary values of a deal, deal owners, and deal stages. <br><br>To update existing deals in your Apollo account, use the <a href=\"https://docs.apollo.io/reference/update-deal\" target=\"_blank\">Update Deal endpoint</a> instead. <br><br>This endpoint also requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [
            {
                name: "name",
                description: "Name the deal you are creating. This should be a human-readable name. <br><br>Example: `Massive Q3 Deal`",
                type: "Query",
                schema: createDealQueryParamsSchema.shape["name"]
            },
            {
                name: "owner_id",
                description: "The ID for the deal owner within your team's Apollo account. <br><br>Use the <a href=\"https://docs.apollo.io/reference/get-a-list-of-users\" target=\"_blank\">Get a List of Users endpoint</a> to retrieve IDs for all of the users within your Apollo account. <br><br>Example: `66302798d03b9601c7934ebf`",
                type: "Query",
                schema: createDealQueryParamsSchema.shape["owner_id"]
            },
            {
                name: "account_id",
                description: "The ID for the account within your Apollo instance. This is the company that you are targeting as part of the deal being created. <br><br>Each company in the Apollo database is assigned a unique ID. To find IDs, call the <a href=\"https://docs.apollo.io/reference/organization-search\" target=\"_blank\">Organization Search endpoint</a> and identify the values for `organization_id`. <br><br>Example: `5e66b6381e05b4008c8331b8`",
                type: "Query",
                schema: createDealQueryParamsSchema.shape["account_id"]
            },
            {
                name: "amount",
                description: "The monetary value of the deal being created. <br><br>Do not enter commas or currency symbols for the value. The currency is automatically populated by the settings within your Apollo account. Commas are not accepted and result in the deal amount being left blank. <br><br>Example: `55123478` (results in a deal value of `$55,123,478` if the default currency is USD)",
                type: "Query",
                schema: createDealQueryParamsSchema.shape["amount"]
            },
            {
                name: "opportunity_stage_id",
                description: "The ID for the deal stage within your team's Apollo account. <br><br>Each deal stage is assigned a unique ID. To find deal stage IDs, call the <a href=\"https://docs.apollo.io/reference/get-a-list-of-opportunity-stages\" target=\"_blank\">List Deal Stages endpoint</a> and identify the value for `id` for each stage. <br><br>Example: `6095a710bd01d100a506d4bd`",
                type: "Query",
                schema: createDealQueryParamsSchema.shape["opportunity_stage_id"]
            },
            {
                name: "closed_date",
                description: "The estimated close date for the deal. This can be a future or past date. <br><br>The date should be formatted as `YYYY-MM-DD`. <br><br>Example: `2025-10-30`",
                type: "Query",
                schema: createDealQueryParamsSchema.shape["closed_date"]
            }
        ],
        response: createDealMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: createDeal401Schema
            },
            {
                status: 403,
                description: "403",
                schema: createDeal403Schema
            },
            {
                status: 422,
                description: "422",
                schema: createDeal422Schema
            },
            {
                status: 429,
                description: "429",
                schema: createDeal429Schema
            }
        ],
        alias: "createOpportunity",
    },
    {
        method: "get",
        path: "/opportunities/search",
        description: "Use the List All Deals endpoint to retrieve every deal that has been created for your team's Apollo account. <br><br>This endpoint also requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key. <br><br>This feature is not accessible to Apollo users on free plans.",
        requestFormat: "json",
        parameters: [
            {
                name: "sort_by_field",
                description: "Sort the tasks by 1 of the following options:    <ul> <li> `amount`: The largest deal values first. </li> <li> `is_closed`: Deals that have been closed first. </li> <li> `is_won`: Deals that have been won first. </li> </ul>",
                type: "Query",
                schema: listAllDealsQueryParamsSchema.shape["sort_by_field"]
            },
            {
                name: "page",
                description: "The page number of the Apollo data that you want to retrieve. <br><br>Use this parameter in combination with the `per_page` parameter to make search results for navigable and improve the performance of the endpoint. <br><br>Example: `4`",
                type: "Query",
                schema: listAllDealsQueryParamsSchema.shape["page"]
            },
            {
                name: "per_page",
                description: "The number of search results that should be returned for each page. Limited the number of results per page improves the endpoint's performance. <br><br>Use the `page` parameter to search the different pages of data. <br><br>Example: `10`",
                type: "Query",
                schema: listAllDealsQueryParamsSchema.shape["per_page"]
            }
        ],
        response: listAllDealsQueryResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: listAllDeals401Schema
            },
            {
                status: 403,
                description: "403",
                schema: listAllDeals403Schema
            },
            {
                status: 429,
                description: "429",
                schema: listAllDeals429Schema
            }
        ],
        alias: "searchOpportunities",
    },
    {
        method: "get",
        path: "/opportunities/:opportunity_id",
        description: "Use the View Deal endpoint to retrieve complete details about a deal within your team's Apollo account. <br><br>Deal information can include the ID of the deal owner, the monetary value of the deal, the deal stage, and general details about the account. <br><br>This endpoint also requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [
            {
                name: "opportunity_id",
                description: "The ID for the deal you want to view. <br><br>Each deal in the Apollo database is assigned a unique ID. To find deal IDs, call the <a href=\"https://docs.apollo.io/reference/list-all-deals\" target=\"_blank\">List All Deals endpoint</a> and identify the value for `id` for the desired deal. <br><br>Example: `66e09ea8e3cfcf01b2208ec7`",
                type: "Path",
                schema: viewDealPathParamsSchema.shape["opportunity_id"]
            }
        ],
        response: viewDealQueryResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: viewDeal401Schema
            },
            {
                status: 403,
                description: "403",
                schema: viewDeal403Schema
            },
            {
                status: 404,
                description: "404",
                schema: viewDeal404Schema
            },
            {
                status: 422,
                description: "422",
                schema: viewDeal422Schema
            },
            {
                status: 429,
                description: "429",
                schema: viewDeal429Schema
            }
        ],
        alias: "getOpportunity",
    },
    {
        method: "patch",
        path: "/opportunities/:opportunity_id",
        description: "Use the Update Deal endpoint to update the details of existing deals within your team's Apollo account. Deals enable you to track account activity, including monetary values of a deal, deal owners, and deal stages. <br><br>To create new deals in your Apollo account, use the <a href=\"https://docs.apollo.io/reference/create-deal\" target=\"_blank\">Create Deal endpoint</a> instead. <br><br>This endpoint also requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [
            {
                name: "opportunity_id",
                description: "The ID for the deal you want to update. <br><br>Each deal in the Apollo database is assigned a unique ID. To find deal IDs, call the <a href=\"https://docs.apollo.io/reference/list-all-deals\" target=\"_blank\">List All Deals endpoint</a> and identify the value for `id` for the desired deal. <br><br>Example: `66e09ea8e3cfcf01b2208ec7`",
                type: "Path",
                schema: updateDealPathParamsSchema.shape["opportunity_id"]
            },
            {
                name: "owner_id",
                description: "The ID for the deal owner within your team's Apollo account. Enter a different ID to update the owner of the deal. <br><br>Use the <a href=\"https://docs.apollo.io/reference/get-a-list-of-users\" target=\"_blank\">Get a List of Users endpoint</a> to retrieve IDs for all of the users within your Apollo account. <br><br>Example: `66302798d03b9601c7934ebf`",
                type: "Query",
                schema: updateDealQueryParamsSchema.shape["owner_id"]
            },
            {
                name: "name",
                description: "Update the name of the deal. This should be a human-readable name. <br><br>Example: `Massive Q3 Deal`",
                type: "Query",
                schema: updateDealQueryParamsSchema.shape["name"]
            },
            {
                name: "amount",
                description: "The monetary value of the deal. Enter a different value to update the deal amount. <br><br>Do not enter commas or currency symbols for the value. The currency is automatically populated by the settings within your Apollo account. Commas are not accepted and result in the deal amount being left blank. <br><br>Example: `55123478` (results in a deal value of `$55,123,478` if the default currency is USD)",
                type: "Query",
                schema: updateDealQueryParamsSchema.shape["amount"]
            },
            {
                name: "opportunity_stage_id",
                description: "The ID for the deal stage within your team's Apollo account. Enter a different ID to update the deal stage. <br><br>Each deal stage is assigned a unique ID. To find deal stage IDs, call the <a href=\"https://docs.apollo.io/reference/get-a-list-of-opportunity-stages\" target=\"_blank\">List Deal Stages endpoint</a> and identify the value for `id` for each stage. <br><br>Example: `6095a710bd01d100a506d4bd`",
                type: "Query",
                schema: updateDealQueryParamsSchema.shape["opportunity_stage_id"]
            },
            {
                name: "closed_date",
                description: "Update the estimated close date for the deal. This can be a future or past date. <br><br>The date should be formatted as `YYYY-MM-DD`. <br><br>Example: `2025-10-30`",
                type: "Query",
                schema: updateDealQueryParamsSchema.shape["closed_date"]
            },
            {
                name: "is_closed",
                description: "Set the value to `true` to update the status of the deal to `closed`.",
                type: "Query",
                schema: updateDealQueryParamsSchema.shape["is_closed"]
            },
            {
                name: "is_won",
                description: "Set the value to `true` to update the status of the deal to `won`.",
                type: "Query",
                schema: updateDealQueryParamsSchema.shape["is_won"]
            },
            {
                name: "source",
                description: "Update the source of the deal. <br><br>By default, deals created via Apollo APIs have a source of `api`. <br><br>Example: `2024 InfoSec Conference`",
                type: "Query",
                schema: updateDealQueryParamsSchema.shape["source"]
            },
            {
                name: "account_id",
                description: "The ID for the account within your Apollo instance. Enter a different ID to update the company associated with the deal. <br><br>Each company in the Apollo database is assigned a unique ID. To find IDs, call the <a href=\"https://docs.apollo.io/reference/organization-search\" target=\"_blank\">Organization Search endpoint</a> and identify the values for `organization_id`. <br><br>Example: `5e66b6381e05b4008c8331b8`",
                type: "Query",
                schema: updateDealQueryParamsSchema.shape["account_id"]
            }
        ],
        response: updateDealMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: updateDeal401Schema
            },
            {
                status: 403,
                description: "403",
                schema: updateDeal403Schema
            },
            {
                status: 422,
                description: "422",
                schema: updateDeal422Schema
            },
            {
                status: 429,
                description: "429",
                schema: updateDeal429Schema
            }
        ],
        alias: "patchOpportunity",
    },
    {
        method: "get",
        path: "/opportunity_stages",
        description: "Use the List Deal Stages endpoint to retrieve information about every deal stage that is available in your team's Apollo account. <br><br>The `id` value for each deal stage can be used to set the stage when <a href=\"https://docs.apollo.io/reference/create-deal\" target=\"_blank\">creating a deal</a> or <a href=\"https://docs.apollo.io/reference/update-deal\" target=\"_blank\">updating a deal</a> via the Apollo API. <br><br>This endpoint also requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [],
        response: listDealStagesQueryResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: listDealStages401Schema
            },
            {
                status: 403,
                description: "403",
                schema: listDealStages403Schema
            },
            {
                status: 429,
                description: "429",
                schema: listDealStages429Schema
            }
        ],
        alias: "listDealStages",
    },
    {
        method: "post",
        path: "/contacts",
        description: "Use the Create a Contact endpoint to add a new contact to your team's Apollo account. <br><br>In Apollo terminology, a contact is a person that your team has explicitly added to your database. A contact will have their data enriched in some way, such as accessing an email address or a phone number. <br><br>Apollo does not apply deduplication processes when you create a new contact via the API. If your entry has the same name, email address, or other details as an existing contact, Apollo will create a new contact instead of updating the existing contact. <br><br>To update an existing contact, use the <a href=\"https://docs.apollo.io/reference/update-a-contact\" target=\"_blank\">Update a Contact endpoint</a> instead.",
        requestFormat: "json",
        parameters: [
            {
                name: "first_name",
                description: "The first name of the contact you want to create. This should be a human-readable name. <br><br>Example: `Tim`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["first_name"]
            },
            {
                name: "last_name",
                description: "The last name of the contact you want to create. This should be a human-readable name. <br><br>Example: `Zheng`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["last_name"]
            },
            {
                name: "organization_name",
                description: "The name of the contact's employer (company). This should be the current employer. <br><br>To ensure a contact is assigned to the right company in the Apollo contact base, check the exact company name using the <a href=\"https://docs.apollo.io/reference/organization-search\" target=\"_blank\">Organization Search endpoint</a>. <br><br>Example: `apollo`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["organization_name"]
            },
            {
                name: "title",
                description: "The current job title that the contact holds. <br><br>Example: `senior research analyst`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["title"]
            },
            {
                name: "account_id",
                description: "The Apollo ID for the account to which you want to assign the contact. Each company with enriched data is assigned a unique account ID. <br><br>To find account IDs, call the <a href=\"https://docs.apollo.io/reference/organization-search\" target=\"_blank\">Organization Search endpoint</a> and identify the value for `id` within the `accounts[]` array for each company. If a company is not already part of an account within your Apollo instance, you instead need to call either the <a href=\"https://docs.apollo.io/reference/organization-enrichment\" target=\"_blank\">Organization Enrichment endpoint</a> or the <a href=\"https://docs.apollo.io/reference/bulk-organization-enrichment\" target=\"_blank\">Bulk Organization Enrichment endpoint</a> to enrich the organization's data, then identify the value for `account_id` for the company. <br><br>Be sure you understand the difference between organization IDs and account IDs. Every company in the Apollo database is assigned an organization ID, but only companies with enriched data are assigned an account ID. If you use an organization ID with this endpoint, you might still receive a `200` response, but the contact will be created without being assigned to the desired account. <br><br>Example: `63f53afe4ceeca00016bdd2f`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["account_id"]
            },
            {
                name: "email",
                description: "The email address of the contact. <br><br>Example: `example@email.com`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["email"]
            },
            {
                name: "website_url",
                description: "The corporate website URL for the contact's current employer (company). <br><br>This should be the full URL, including `https://www.`, `.com`, and similar. Do not add URL slugs or sub-directories. <br><br>Do not use social media URLs, such as LinkedIn profile links. This will lead to inaccurate data enrichment for the contact. <br><br>Examples: `https://www.apollo.io/`; `https://www.microsoft.com/`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["website_url"]
            },
            {
                name: "label_names[]",
                description: "Add the contact to lists within your team's Apollo account. If your entry does not match the name of a list within your account, Apollo automatically creates a new list. <br><br>Lists enable your team to consolidate related contacts and accounts so you can more easily prospect and engage them. When in the Apollo product (not API), users can use lists as a search filter, add lists to a sequence to automate outreach, and add lists to workflows for task scheduling and sequence maintenance. <br><br>Examples: `2024 big marketing conference attendees`; `inbound contact`; `smb clients`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["label_names[]"]
            },
            {
                name: "contact_stage_id",
                description: "The Apollo ID for the contact stage to which you want to assign the contact. Call the <a href=\"https://docs.apollo.io/reference/list-contact-stages\" target=\"_blank\">List Contact Stages endpoint</a> to retrieve a list of all the contact stage IDs available in your Apollo account.<br><br>If you do not specify the contact stage, Apollo automatically assigns the contact to a stage as determined by your team's Apollo account. To change the order of contact stages, launch the Apollo product and go to <b>Settings</b> > <b>Objects</b> > <a href=\"https://app.apollo.io/#/settings/contacts/stages\" target=\"_blank\"><b>Contacts</b></a>. Then, access the <b>Triggers</b> tab and change default stage for when a prospect is created.<br><br>Example: `6095a710bd01d100a506d4ae`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["contact_stage_id"]
            },
            {
                name: "present_raw_address",
                description: "The personal location for the contact. This can include a city, US state, and country. <br><br>Apollo matches the location you provide to the most applicable pre-defined location. <br><br>Examples: `Atlanta, United States`; `Tokyo, Japan`; `Saint Petersburg, Russia`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["present_raw_address"]
            },
            {
                name: "direct_phone",
                description: "The primary phone number for the contact. <br><br>Apollo sanitizes phone numbers, so you can enter them in any format. The sanitized number can be viewed in the endpoint response. <br><br>Examples: `555-303-1234`; `+44 7911 123456`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["direct_phone"]
            },
            {
                name: "corporate_phone",
                description: "The work/office phone number for the contact. <br><br>This is a direct line to the contact at their employer, not the corporate headquarters phone number. <br><br>Apollo sanitizes phone numbers, so you can enter them in any format. The sanitized number can be viewed in the endpoint response. <br><br>Examples: `555-303-1234`; `+44 7911 123456`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["corporate_phone"]
            },
            {
                name: "mobile_phone",
                description: "The mobile phone number for the contact. <br><br>Apollo sanitizes phone numbers, so you can enter them in any format. The sanitized number can be viewed in the endpoint response. <br><br>Examples: `555-303-1234`; `+44 7911 123456`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["mobile_phone"]
            },
            {
                name: "home_phone",
                description: "The home phone number for the contact. <br><br>Apollo sanitizes phone numbers, so you can enter them in any format. The sanitized number can be viewed in the endpoint response. <br><br>Examples: `555-303-1234`; `+44 7911 123456`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["home_phone"]
            },
            {
                name: "other_phone",
                description: "An unknown type of phone number or an alternative phone number for the contact. <br><br>Apollo sanitizes phone numbers, so you can enter them in any format. The sanitized number can be viewed in the endpoint response. <br><br>Examples: `555-303-1234`; `+44 7911 123456`",
                type: "Query",
                schema: createAContactQueryParamsSchema.shape["other_phone"]
            }
        ],
        response: createAContactMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: createAContact401Schema
            },
            {
                status: 422,
                description: "422",
                schema: createAContact422Schema
            },
            {
                status: 429,
                description: "429",
                schema: createAContact429Schema
            }
        ],
        alias: "createContact",
    },
    {
        method: "put",
        path: "/contacts/:contact_id",
        description: "Use the Update a Contact endpoint to update existing contacts in your team's Apollo account. <br><br>In Apollo terminology, a contact is a person that your team has explicitly added to your database. A contact will have their data enriched in some way, such as accessing an email address or a phone number. <br><br>To create a new contact, use the <a href=\"https://docs.apollo.io/reference/create-a-contact\" target=\"_blank\">Create a Contact endpoint</a> instead. To update the contact stage for multiple contacts, use the <a href=\"https://docs.apollo.io/reference/update-contact-stage\" target=\"_blank\">Update Contact Stage for Multiple Contacts</a> endpoint.",
        requestFormat: "json",
        parameters: [
            {
                name: "contact_id",
                description: "The Apollo ID for the contact that you want to update. <br><br>To find contact IDs, call the <a href=\"https://docs.apollo.io/reference/search-for-contacts\" target=\"_blank\">Search for Contacts endpoint</a> and identify the `id` value for the contact. <br><br>Example: `66e34b81740c50074e3d1bd4`",
                type: "Path",
                schema: updateAContactPathParamsSchema.shape["contact_id"]
            },
            {
                name: "first_name",
                description: "Update the contact's first name. This should be a human-readable name. <br><br>Example: `Tim`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["first_name"]
            },
            {
                name: "last_name",
                description: "Update the contact's last name. This should be a human-readable name. <br><br>Example: `Zheng`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["last_name"]
            },
            {
                name: "organization_name",
                description: "Update the name of the contact's employer (company). This should be the current employer. <br><br>To ensure a contact is assigned to the right company in the Apollo contact base, check the exact company name using the <a href=\"https://docs.apollo.io/reference/organization-search\" target=\"_blank\">Organization Search endpoint</a>. <br><br>Example: `apollo`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["organization_name"]
            },
            {
                name: "title",
                description: "Update the job title that the contact holds. <br><br>Example: `senior research analyst`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["title"]
            },
            {
                name: "account_id",
                description: "The Apollo ID to update the account to which the contact is assigned. Each company with enriched data is assigned a unique account ID. <br><br>To find account IDs, call the <a href=\"https://docs.apollo.io/reference/organization-search\" target=\"_blank\">Organization Search endpoint</a> and identify the value for `id` within the `accounts[]` array for each company. If a company is not already part of an account within your Apollo instance, you instead need to call either the <a href=\"https://docs.apollo.io/reference/organization-enrichment\" target=\"_blank\">Organization Enrichment endpoint</a> or the <a href=\"https://docs.apollo.io/reference/bulk-organization-enrichment\" target=\"_blank\">Bulk Organization Enrichment endpoint</a> to enrich the organization's data, then identify the value for `account_id` for the company. <br><br>Be sure you understand the difference between organization IDs and account IDs. Every company in the Apollo database is assigned an organization ID, but only companies with enriched data are assigned an account ID. If you use an organization ID with this endpoint, you might still receive a `200` response, but the contact will be created without being assigned to the desired account. <br><br>Example: `63f53afe4ceeca00016bdd2f`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["account_id"]
            },
            {
                name: "email",
                description: "Update the email address of the contact. <br><br>Example: `example@email.com`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["email"]
            },
            {
                name: "website_url",
                description: "Update the corporate website URL for the contact's current employer (company). <br><br>This should be the full URL, including `https://www.`, `.com`, and similar. Do not add URL slugs or sub-directories. <br><br>Do not use social media URLs, such as LinkedIn profile links. This will lead to inaccurate data enrichment for the contact. <br><br>Examples: `https://www.apollo.io/`; `https://www.microsoft.com/`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["website_url"]
            },
            {
                name: "label_names[]",
                description: "Update the lists that the contact belongs to within your team's Apollo account. If your entry does not match the name of a list within your account, Apollo automatically creates a new list. <br><br>If you use this parameter to add the contact to lists, Apollo automatically removes them from any existing lists to which they were already added, unless you add those lists again. <br><br>Lists enable your team to consolidate related contacts and accounts so you can more easily prospect and engage them. When in the Apollo product (not API), users can use lists as a search filter, add lists to a sequence to automate outreach, and add lists to workflows for task scheduling and sequence maintenance. <br><br>Examples: `2024 big marketing conference attendees`; `inbound contact`; `smb clients`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["label_names[]"]
            },
            {
                name: "contact_stage_id",
                description: "The Apollo ID to update the contact stage to which the contact is assigned. Call the <a href=\"https://docs.apollo.io/reference/list-contact-stages\" target=\"_blank\">List Contact Stages endpoint</a> to retrieve a list of all the contact stage IDs available in your Apollo account. <br><br>Example: `6095a710bd01d100a506d4af`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["contact_stage_id"]
            },
            {
                name: "present_raw_address",
                description: "Update the personal location for the contact. This can include a city, US state, and country. <br><br>Apollo matches the location you provide to the most applicable pre-defined location. <br><br>Examples: `Atlanta, United States`; `Tokyo, Japan`; `Saint Petersburg, Russia`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["present_raw_address"]
            },
            {
                name: "direct_phone",
                description: "Update the primary phone number for the contact. <br><br>Apollo sanitizes phone numbers, so you can enter them in any format. The sanitized number can be viewed in the endpoint response. <br><br>Examples: `555-303-1234`; `+44 7911 123456`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["direct_phone"]
            },
            {
                name: "corporate_phone",
                description: "Update the work/office phone number for the contact. <br><br>This is a direct line to the contact at their employer, not the corporate headquarters phone number. <br><br>Apollo sanitizes phone numbers, so you can enter them in any format. The sanitized number can be viewed in the endpoint response. <br><br>Examples: `555-303-1234`; `+44 7911 123456`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["corporate_phone"]
            },
            {
                name: "mobile_phone",
                description: "Update the mobile phone number for the contact. <br><br>Apollo sanitizes phone numbers, so you can enter them in any format. The sanitized number can be viewed in the endpoint response. <br><br>Examples: `555-303-1234`; `+44 7911 123456`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["mobile_phone"]
            },
            {
                name: "home_phone",
                description: "Update the home phone number for the contact. <br><br>Apollo sanitizes phone numbers, so you can enter them in any format. The sanitized number can be viewed in the endpoint response. <br><br>Examples: `555-303-1234`; `+44 7911 123456`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["home_phone"]
            },
            {
                name: "other_phone",
                description: "Update an unknown type of phone number or an alternative phone number for the contact. <br><br>Apollo sanitizes phone numbers, so you can enter them in any format. The sanitized number can be viewed in the endpoint response. <br><br>Examples: `555-303-1234`; `+44 7911 123456`",
                type: "Query",
                schema: updateAContactQueryParamsSchema.shape["other_phone"]
            }
        ],
        response: updateAContactMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: updateAContact401Schema
            },
            {
                status: 422,
                description: "422",
                schema: updateAContact422Schema
            },
            {
                status: 429,
                description: "429",
                schema: updateAContact429Schema
            }
        ],
        alias: "updateContact",
    },
    {
        method: "post",
        path: "/contacts/search",
        description: "Use the Search for Contacts endpoint to search for the contacts that have been added to your team's Apollo account. <br><br>In Apollo terminology, a contact is a person that your team has explicitly added to your database. A contact will have their data enriched in some way, such as accessing an email address or a phone number. <br><br>This endpoint only returns contacts in the search results. To search for people in the Apollo database, call the <a href=\"https://docs.apollo.io/reference/people-search\" target=\"_blank\">People Search endpoint</a>. <br><br>To protect Apollo's performance for all users, this endpoint has a display limit of 50,000 records (100 records per page, up to 500 pages). Add more filters to narrow your search results as much as possible. This limitation does not restrict your access to Apollo's database; you just need to access the data in batches. <br><br>This feature is not accessible to Apollo users on free plans.",
        requestFormat: "json",
        parameters: [
            {
                name: "q_keywords",
                description: "Add keywords to narrow the search of the contacts in your team's Apollo account. <br><br>Keywords can include combinations of names, job titles, employers (company names), and email addresses. <br><br>Examples: `tim zheng`; `senior research analyst`; `microsoft`",
                type: "Query",
                schema: searchForContactsQueryParamsSchema.shape["q_keywords"]
            },
            {
                name: "contact_stage_ids[]",
                description: "The Apollo IDs for the contact stages that you want to include in your search results. If you add multiple contact stages, Apollo will include all contacts that match any of the stages, along with the other parameters, in the search results. <br><br>Call the <a href=\"https://docs.apollo.io/reference/list-contact-stages\" target=\"_blank\">List Contact Stages endpoint</a> to retrieve a list of all the contact stage IDs available in your Apollo account. <br>Example: `6095a710bd01d100a506d4ae`",
                type: "Query",
                schema: searchForContactsQueryParamsSchema.shape["contact_stage_ids[]"]
            },
            {
                name: "sort_by_field",
                description: "Sort the matching contacts by 1 of the following options: <ul> <li> `contact_last_activity_date`: The most recent activity date recorded first. </li> <li> `contact_email_last_opened_at`: The most recent email opened date first. </li> <li> `contact_email_last_clicked_at`: The most recent email clicked first. </li> <li> `contact_created_at`: The most recently created first. </li> <li> `contact_updated_at`: The most recently updated first. </li> </ul>",
                type: "Query",
                schema: searchForContactsQueryParamsSchema.shape["sort_by_field"]
            },
            {
                name: "sort_ascending",
                description: "Set to `true` to sort the matching contacts in ascending order. <br><br>This parameter must be used with `sort_by_field`. Otherwise, the sorting logic is not applied. <br><br>Example: `true`",
                type: "Query",
                schema: searchForContactsQueryParamsSchema.shape["sort_ascending"]
            },
            {
                name: "per_page",
                description: "The page number of the Apollo data that you want to retrieve. <br><br>Use this parameter in combination with the `per_page` parameter to make search results for navigable and improve the performance of the endpoint. <br><br>Example: `4`",
                type: "Query",
                schema: searchForContactsQueryParamsSchema.shape["per_page"]
            },
            {
                name: "page",
                description: "The number of search results that should be returned for each page. Limited the number of results per page improves the endpoint's performance. <br><br>Use the `page` parameter to search the different pages of data. <br><br>Example: `10`",
                type: "Query",
                schema: searchForContactsQueryParamsSchema.shape["page"]
            }
        ],
        response: searchForContactsMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: searchForContacts401Schema
            },
            {
                status: 403,
                description: "403",
                schema: searchForContacts403Schema
            },
            {
                status: 429,
                description: "429",
                schema: searchForContacts429Schema
            }
        ],
        alias: "searchContacts",
    },
    {
        method: "get",
        path: "/contact_stages",
        description: "Use the List Contact Stages endpoint to retrieve the IDs for the available contact stages in your team's Apollo account. <br><br>Contact stage IDs can be used to <a href=\"https://docs.apollo.io/reference/update-a-contact\" target=\"_blank\">update invidual contacts</a> and <a href=\"https://docs.apollo.io/reference/update-contact-stage\" target=\"_blank\">update the contact stages for multiple contacts</a> via the Apollo API.",
        requestFormat: "json",
        parameters: [],
        response: listContactStagesQueryResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: listContactStages401Schema
            },
            {
                status: 403,
                description: "403",
                schema: listContactStages403Schema
            },
            {
                status: 429,
                description: "429",
                schema: listContactStages429Schema
            }
        ],
        alias: "listContactStages",
    },
    {
        method: "post",
        path: "/contacts/update_stages",
        description: "Use the Update Contact Stage for Multiple Contacts endpoint to update the contact stage for several contacts in your team's Apollo account. <br><br>To update more than the contact stage, such as job titles or contact details, use the <a href=\"https://docs.apollo.io/reference/update-a-contact\" target=\"_blank\">Update a Contact endpoint</a> instead.",
        requestFormat: "json",
        parameters: [
            {
                name: "contact_ids[]",
                description: "The Apollo IDs for the contacts that you want to update. <br><br>To find contact IDs, call the <a href=\"https://docs.apollo.io/reference/search-for-contacts\" target=\"_blank\">Search for Contacts endpoint</a> and identify the `id` value for the contact. <br><br>Example: `66e34b81740c50074e3d1bd4`",
                type: "Query",
                schema: updateContactStageQueryParamsSchema.shape["contact_ids[]"]
            },
            {
                name: "contact_stage_id",
                description: "The Apollo ID for the contact stage to which you want to assign the contacts. Call the <a href=\"https://docs.apollo.io/reference/list-contact-stages\" target=\"_blank\">List Contact Stages endpoint</a> to retrieve a list of all the contact stage IDs available in your Apollo account. <br><br>Example: `6095a710bd01d100a506d4af`",
                type: "Query",
                schema: updateContactStageQueryParamsSchema.shape["contact_stage_id"]
            }
        ],
        response: updateContactStageMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: updateContactStage401Schema
            },
            {
                status: 422,
                description: "422",
                schema: updateContactStage422Schema
            },
            {
                status: 429,
                description: "429",
                schema: updateContactStage429Schema
            }
        ],
        alias: "updateContactStage",
    },
    {
        method: "post",
        path: "/contacts/update_owners",
        description: "Use the Update Contact Owner for Multiple Contacts endpoint to assign multiple contacts to a different user in your team's Apollo account. <br><br>To update more than the contact's owner, such as job titles or contact details, use the <a href=\"https://docs.apollo.io/reference/update-a-contact\" target=\"_blank\">Update a Contact endpoint</a> instead.",
        requestFormat: "json",
        parameters: [
            {
                name: "contact_ids[]",
                description: "The Apollo IDs for the contacts that you want assign to an owner. <br><br>To find contact IDs, call the <a href=\"https://docs.apollo.io/reference/search-for-contacts\" target=\"_blank\">Search for Contacts endpoint</a> and identify the `id` value for the contact. <br><br>Example: `66e34b81740c50074e3d1bd4`",
                type: "Query",
                schema: updateContactOwnershipQueryParamsSchema.shape["contact_ids[]"]
            },
            {
                name: "owner_id",
                description: "The ID for the contact owner within your team's Apollo account. This user will be assigned ownership of the contacts. <br><br>Use the <a href=\"https://docs.apollo.io/reference/get-a-list-of-users\" target=\"_blank\">Get a List of Users endpoint</a> endpoint to retrieve IDs for all of the users within your Apollo account. <br><br>Example: `66302798d03b9601c7934ebf`",
                type: "Query",
                schema: updateContactOwnershipQueryParamsSchema.shape["owner_id"]
            }
        ],
        response: updateContactOwnershipMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: updateContactOwnership401Schema
            },
            {
                status: 422,
                description: "422",
                schema: updateContactOwnership422Schema
            },
            {
                status: 429,
                description: "429",
                schema: updateContactOwnership429Schema
            }
        ],
        alias: "updateContactOwnership",
    },
    {
        method: "post",
        path: "/accounts",
        description: "Use the Create an Account endpoint to add a new account to your team's Apollo account. <br><br>In Apollo terminology, an account is a company that your team has explicitly added to your database. <br><br>Apollo does not apply deduplication processes when you create a new account via the API. If your entry has the same name, domain, or other details as an existing account, Apollo will create a new account instead of updating the existing account. To update an existing account, use the <a href=\"https://docs.apollo.io/reference/update-an-account\" target=\"_blank\">Update an Account endpoint</a> instead. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [
            {
                name: "name",
                description: "Name the account that you are creating. This should be a human-readable name. <br><br>Example: `The Irish Copywriters`",
                type: "Query",
                schema: createAnAccountQueryParamsSchema.shape["name"]
            },
            {
                name: "domain",
                description: "The domain name for the account. <br><br>Do not include `www.` or similar. <br><br>Example: `apollo.io` or `microsoft.com`",
                type: "Query",
                schema: createAnAccountQueryParamsSchema.shape["domain"]
            },
            {
                name: "owner_id",
                description: "The ID for the account owner within your team's Apollo account. <br><br>Use the <a href=\"https://docs.apollo.io/reference/get-a-list-of-users\" target=\"_blank\">Get a List of Users endpoint</a> to retrieve IDs for all of the users within your Apollo account. <br><br>Example: `66302798d03b9601c7934ebf`",
                type: "Query",
                schema: createAnAccountQueryParamsSchema.shape["owner_id"]
            },
            {
                name: "account_stage_id",
                description: "The Apollo ID for the account stage to which you want to assign the account. Call the <a href=\"https://docs.apollo.io/reference/list-account-stages\" target=\"_blank\">List Account Stages endpoint</a> to retrieve a list of all the account stage IDs available in your Apollo account. <br><br>If you do not specify the account stage, Apollo automatically assigns the account to a stage as determined by your team's Apollo account. To change the order of account stages, launch the Apollo product and go to <b>Settings</b> > <b>Objects</b> > <a href=\"https://app.apollo.io/#/settings/accounts/stages\" target=\"_blank\"><b>Accounts</b></a>. Then, access the <b>Triggers</b> tab and change the stage for when an account is created.<br><br>Example: `6095a710bd01d100a506d4b9`",
                type: "Query",
                schema: createAnAccountQueryParamsSchema.shape["account_stage_id"]
            },
            {
                name: "phone",
                description: "The primary phone number for the account. <br><br>This can be the phone number for the corporate headquarters, a branch location, or a direct dial to the primary point of contact for the account. <br><br>Apollo sanitizes phone numbers, so you can enter them in any format. The sanitized number can be viewed in the endpoint response. <br><br>Examples: `555-303-1234`; `+44 7911 123456`",
                type: "Query",
                schema: createAnAccountQueryParamsSchema.shape["phone"]
            },
            {
                name: "raw_address",
                description: "The corporate location for the account. This can include a city, US state, and country. <br><br>Apollo matches the location you provide to the most applicable pre-defined location. <br><br>Examples: `Belfield, Dublin 4, Ireland`; `Dallas, United States`",
                type: "Query",
                schema: createAnAccountQueryParamsSchema.shape["raw_address"]
            }
        ],
        response: createAnAccountMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: createAnAccount401Schema
            },
            {
                status: 403,
                description: "403",
                schema: createAnAccount403Schema
            },
            {
                status: 422,
                description: "422",
                schema: createAnAccount422Schema
            },
            {
                status: 429,
                description: "429",
                schema: createAnAccount429Schema
            }
        ],
        alias: "createAccount",
    },
    {
        method: "put",
        path: "/accounts/:account_id",
        description: "Use the Update an Account endpoint to update existing accounts in your team's Apollo account. <br><br>In Apollo terminology, an account is a company that your team has explicitly added to your database. <br><br>To create a new account, use the <a href=\"https://docs.apollo.io/reference/create-an-account\" target=\"_blank\">Create an Account endpoint</a> instead. To update the account stage for multiple account, use the <a href=\"https://docs.apollo.io/reference/update-account-stage\" target=\"_blank\">Update Account Stage for Multiple Accounts endpoint</a>. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [
            {
                name: "account_id",
                description: "The Apollo ID for the account that you want to update. <br><br>To find account IDs, call the <a href=\"https://docs.apollo.io/reference/search-for-accounts\" target=\"_blank\">Search for Accounts endpoint</a> and identify the `id` value for the contact. <br><br>Example: `66e9abf95ac32901b20d1a0d`",
                type: "Path",
                schema: updateAnAccountPathParamsSchema.shape["account_id"]
            },
            {
                name: "name",
                description: "Update the account's name. This should be a human-readable name. <br><br>Example: `The Fast Irish Copywriters`",
                type: "Query",
                schema: updateAnAccountQueryParamsSchema.shape["name"]
            },
            {
                name: "domain",
                description: "Update the domain name for the account.  Do not include `www.` or similar. <br><br>Example: `apollo.io` or `microsoft.com`",
                type: "Query",
                schema: updateAnAccountQueryParamsSchema.shape["domain"]
            },
            {
                name: "owner_id",
                description: "The ID for the account owner within your team's Apollo account. Enter a different ID to update the owner of the account. <br><br>Use the <a href=\"https://docs.apollo.io/reference/get-a-list-of-users\" target=\"_blank\">Get a List of Users endpoint</a> to retrieve IDs for all of the users within your Apollo account. <br><br>Example: `66302798d03b9601c7934ebf`",
                type: "Query",
                schema: updateAnAccountQueryParamsSchema.shape["owner_id"]
            },
            {
                name: "account_stage_id",
                description: "The Apollo ID for the account stage to which you want to assign the account. Enter a different ID to update the account stage. <br><br>Call the <a href=\"https://docs.apollo.io/reference/list-account-stages\" target=\"_blank\">List Account Stages endpoint</a> to retrieve a list of all the account stage IDs available in your Apollo account. <br><br>If you do not specify the account stage, Apollo automatically assigns the account to a stage as determined by your team's Apollo account. To change the order of account stages, launch the Apollo product and go to <b>Settings</b> > <b>Objects</b> > <a href=\"https://app.apollo.io/#/settings/accounts/stages\" target=\"_blank\"><b>Accounts</b></a>. Then, access the <b>Triggers</b> tab and change the stage for when an account is created. <br><br>Example: `61b8e913e0f4d2012e3af74e`",
                type: "Query",
                schema: updateAnAccountQueryParamsSchema.shape["account_stage_id"]
            },
            {
                name: "raw_address",
                description: "Update the corporate location for the account. This can include a city, US state, and country. <br><br>Apollo matches the location you provide to the most applicable pre-defined location. <br><br>Examples: `Belfield, Dublin 4, Ireland`; `Dallas, United States`",
                type: "Query",
                schema: updateAnAccountQueryParamsSchema.shape["raw_address"]
            },
            {
                name: "phone",
                description: "Update the primary phone number for the account. <br><br>This can be the phone number for the corporate headquarters, a branch location, or a direct dial to the primary point of contact for the account. <br><br>Apollo sanitizes phone numbers, so you can enter them in any format. The sanitized number can be viewed in the endpoint response. <br><br>Examples: `555-303-1234`; `+44 7911 123456`",
                type: "Query",
                schema: updateAnAccountQueryParamsSchema.shape["phone"]
            }
        ],
        response: updateAnAccountMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: updateAnAccount401Schema
            },
            {
                status: 403,
                description: "403",
                schema: updateAnAccount403Schema
            },
            {
                status: 422,
                description: "422",
                schema: updateAnAccount422Schema
            },
            {
                status: 429,
                description: "429",
                schema: updateAnAccount429Schema
            }
        ],
        alias: "updateAccount",
    },
    {
        method: "post",
        path: "/accounts/search",
        description: "Use the Search for Accounts endpoint to search for the account that have been added to your team's Apollo account. <br><br>In Apollo terminology, an account is a company that your team has explicitly added to your database. <br><br>This endpoint only returns accounts in the search results. To search for companies in the Apollo database, call the <a href=\"https://docs.apollo.io/reference/organization-search\" target=\"_blank\">Organization Search endpoint</a>. <br><br>To protect Apollo's performance for all users, this endpoint has a display limit of 50,000 records (100 records per page, up to 500 pages). Add more filters to narrow your search results as much as possible. This limitation does not restrict your access to Apollo's database; you just need to access the data in batches. <br><br>This feature is not accessible to Apollo users on free plans.",
        requestFormat: "json",
        parameters: [
            {
                name: "q_organization_name",
                description: "Add keywords to narrow the search of the accounts in your team's Apollo account. <br><br>Keywords should directly match at least part of an account's name. For example, searching the keyword `marketing` might return the result `NY Marketing Unlimited`, but not `NY Market Analysts`. <br><br>This parameter only searches account names, not other account fields. <br><br>Examples: `apollo`; `microsoft`; `marketing`",
                type: "Query",
                schema: searchForAccountsQueryParamsSchema.shape["q_organization_name"]
            },
            {
                name: "account_stage_ids[]",
                description: "The Apollo IDs for the account stages that you want to include in your search results. If you add multiple account stages, Apollo will include all accounts that match any of the stages, along with the other parameters, in the search results. <br><br>Call the <a href=\"https://docs.apollo.io/reference/list-account-stages\" target=\"_blank\">List Account Stages endpoint</a> to retrieve a list of all the account stage IDs available in your Apollo account. <br><br>Example: `61b8e913e0f4d2012e3af74e`",
                type: "Query",
                schema: searchForAccountsQueryParamsSchema.shape["account_stage_ids[]"]
            },
            {
                name: "sort_by_field",
                description: "Sort the matching accounts by 1 of the following options:   <ul> <li> `account_last_activity_date`: The most recent activity date recorded first. </li> <li> `account_created_at`: The most recently created first. </li> <li> `account_updated_at`: The most recently updated first. </li> </ul>",
                type: "Query",
                schema: searchForAccountsQueryParamsSchema.shape["sort_by_field"]
            },
            {
                name: "sort_ascending",
                description: "Set to `true` to sort the matching contacts in ascending order. <br><br>This parameter must be used with `sort_by_field`. Otherwise, the sorting logic is not applied. <br><br>Example: `true`",
                type: "Query",
                schema: searchForAccountsQueryParamsSchema.shape["sort_ascending"]
            },
            {
                name: "page",
                description: "The page number of the Apollo data that you want to retrieve. <br><br>Use this parameter in combination with the `per_page` parameter to make search results for navigable and improve the performance of the endpoint. <br><br>Example: `4`",
                type: "Query",
                schema: searchForAccountsQueryParamsSchema.shape["page"]
            },
            {
                name: "per_page",
                description: "The number of search results that should be returned for each page. Limited the number of results per page improves the endpoint's performance. <br><br>Use the `page` parameter to search the different pages of data. <br><br>Example: `10`",
                type: "Query",
                schema: searchForAccountsQueryParamsSchema.shape["per_page"]
            }
        ],
        response: searchForAccountsMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: searchForAccounts401Schema
            },
            {
                status: 403,
                description: "403",
                schema: searchForAccounts403Schema
            },
            {
                status: 422,
                description: "422",
                schema: searchForAccounts422Schema
            },
            {
                status: 429,
                description: "429",
                schema: searchForAccounts429Schema
            }
        ],
        alias: "searchAccounts",
    },
    {
        method: "get",
        path: "/account_stages",
        description: "Use the List Accounts Stages endpoint to retrieve the IDs for the available account stages in your team's Apollo account. This endpoint does not require parameters. <br><br>Account stage IDs can be used to <a href=\"https://docs.apollo.io/reference/update-an-account\" target=\"_blank\">update individual accounts</a> and <a href=\"https://docs.apollo.io/reference/update-account-stage\" target=\"_blank\">update the account stages for multiple accounts</a> via the Apollo API. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [],
        response: listAccountStagesQueryResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: listAccountStages401Schema
            },
            {
                status: 403,
                description: "403",
                schema: listAccountStages403Schema
            },
            {
                status: 429,
                description: "429",
                schema: listAccountStages429Schema
            }
        ],
        alias: "listAccountStages",
    },
    {
        method: "post",
        path: "/accounts/bulk_update",
        description: "Use the Update Account Stage for Multiple Accounts endpoint to update the account stage for several account in your team's Apollo account. <br><br>To update more than the account stage, such as domains or phone numbers, use the <a href=\"https://docs.apollo.io/reference/update-an-account\" target=\"_blank\">Update an Account endpoint</a> instead. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [
            {
                name: "account_ids[]",
                description: "The Apollo ID for the account that you want to update. <br><br>To find account IDs, call the <a href=\"https://docs.apollo.io/reference/search-for-accounts\" target=\"_blank\">Search for Accounts endpoint</a> and identify the `id` value for the account. <br><br>Example: `66e9abf95ac32901b20d1a0d`",
                type: "Query",
                schema: updateAccountStageQueryParamsSchema.shape["account_ids[]"]
            },
            {
                name: "account_stage_id",
                description: "The Apollo ID for the account stage to which you want to assign the accounts. <br><br>Call the <a href=\"https://docs.apollo.io/reference/list-account-stages\" target=\"_blank\">List Account Stages endpoint</a> to retrieve a list of all the account stage IDs available in your Apollo account. <br><br>Example: `6095a710bd01d100a506d4b7`",
                type: "Query",
                schema: updateAccountStageQueryParamsSchema.shape["account_stage_id"]
            }
        ],
        response: updateAccountStageMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: updateAccountStage401Schema
            },
            {
                status: 403,
                description: "403",
                schema: updateAccountStage403Schema
            },
            {
                status: 429,
                description: "429",
                schema: updateAccountStage429Schema
            }
        ],
        alias: "updateAccountsStage",
    },
    {
        method: "post",
        path: "/accounts/update_owners",
        description: "Use the Update Account Owner for Multiple Accounts endpoint to assign multiple accounts to a different user in your team's Apollo account. <br><br>To update more than the account owner, such as domains or phone numbers, use the <a href=\"https://docs.apollo.io/reference/update-an-account\" target=\"_blank\">Update an Account endpoint</a> instead. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [
            {
                name: "account_ids[]",
                description: "The Apollo IDs for the account that you want to assign to an owner. <br><br>To find account IDs, call the <a href=\"https://docs.apollo.io/reference/search-for-accounts\" target=\"_blank\">Search for Accounts endpoint</a> and identify the `id` value for the account. <br><br>Example: `66e9abf95ac32901b20d1a0d`",
                type: "Query",
                schema: updateAccountOwnershipQueryParamsSchema.shape["account_ids[]"]
            },
            {
                name: "owner_id",
                description: "The ID for the account owner within your team's Apollo account. This user will be assigned ownership of the accounts. <br><br>Use the <a href=\"https://docs.apollo.io/reference/get-a-list-of-users\" target=\"_blank\">Get a List of Users endpoint</a> to retrieve IDs for all of the users within your Apollo account. <br><br>Example: `66302798d03b9601c7934ebf`",
                type: "Query",
                schema: updateAccountOwnershipQueryParamsSchema.shape["owner_id"]
            }
        ],
        response: updateAccountOwnershipMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: updateAccountOwnership401Schema
            },
            {
                status: 403,
                description: "403",
                schema: updateAccountOwnership403Schema
            },
            {
                status: 422,
                description: "422",
                schema: updateAccountOwnership422Schema
            },
            {
                status: 429,
                description: "429",
                schema: updateAccountOwnership429Schema
            }
        ],
        alias: "updateAccountsOwnership",
    },
    {
        method: "post",
        path: "/emailer_campaigns/search",
        description: "Use the Search for Sequences endpoint to search for the sequences that have been created for your team's Apollo account. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key. <br><br>This feature is not accessible to Apollo users on free plans.",
        requestFormat: "json",
        parameters: [
            {
                name: "q_name",
                description: "Add keywords to narrow the search of the sequences in your team's Apollo account. <br><br>Keywords should directly match at least part of a sequence's name. For example, searching the keyword `marketing` might return the result `NY Marketing Sequence`, but not `NY Marketer Conference 2025 attendees`. <br><br>This parameter only searches sequence names, not other sequence fields. <br><br>Example: `marketing conference attendees`",
                type: "Query",
                schema: searchForSequencesQueryParamsSchema.shape["q_name"]
            },
            {
                name: "page",
                description: "The page number of the Apollo data that you want to retrieve. <br><br>Use this parameter in combination with the `per_page` parameter to make search results for navigable and improve the performance of the endpoint. <br><br>Example: `4`",
                type: "Query",
                schema: searchForSequencesQueryParamsSchema.shape["page"]
            },
            {
                name: "per_page",
                description: "The number of search results that should be returned for each page. Limited the number of results per page improves the endpoint's performance. <br><br>Use the `page` parameter to search the different pages of data. <br><br>Example: `10`",
                type: "Query",
                schema: searchForSequencesQueryParamsSchema.shape["per_page"]
            }
        ],
        response: searchForSequencesMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: searchForSequences401Schema
            },
            {
                status: 403,
                description: "403",
                schema: searchForSequences403Schema
            },
            {
                status: 429,
                description: "429",
                schema: searchForSequences429Schema
            }
        ],
        alias: "searchEmailerCampaigns",
    },
    {
        method: "post",
        path: "/emailer_campaigns/:sequence_id/add_contact_ids",
        description: "Use the Add Contacts to a Sequence endpoint to add contacts to the existing sequences in your team's Apollo account. <br><br>In Apollo terminology, a contact is a person that your team has explicitly added to your database. Only contacts can be added to sequences. To enrich a person's data so that they become a contact, call the <a href=\"https://docs.apollo.io/reference/people-enrichment\" target=\"_blank\">People Enrichment endpoint</a>. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [
            {
                name: "sequence_id",
                description: "The Apollo ID for the sequence to which you want to add contacts. <br><br>To find sequence IDs, call the <a href=\"https://docs.apollo.io/reference/search-for-sequences\" target=\"_blank\">Search for Sequences endpoint</a> and identify the `id` value for the sequence. <br><br>Example: `66e9e215ece19801b219997f`",
                type: "Path",
                schema: addContactsToSequencePathParamsSchema.shape["sequence_id"]
            },
            {
                name: "emailer_campaign_id",
                description: "The same ID as the `sequence_id`. <br><br>Example: `66e9e215ece19801b219997f`",
                type: "Query",
                schema: addContactsToSequenceQueryParamsSchema.shape["emailer_campaign_id"]
            },
            {
                name: "contact_ids[]",
                description: "The Apollo IDs for the contacts that you want to add to the sequence. <br><br>To find contact IDs, call the <a href=\"https://docs.apollo.io/reference/search-for-contacts\" target=\"_blank\">Search for Contacts endpoint</a> and identify the `id` value for the contact. <br><br>Example: `66e34b81740c50074e3d1bd4`",
                type: "Query",
                schema: addContactsToSequenceQueryParamsSchema.shape["contact_ids[]"]
            },
            {
                name: "send_email_from_email_account_id",
                description: "The Apollo ID for the email account that you want to use send emails to the contacts you are adding to the sequence. <br><br>To find email account IDs, call the <a href=\"https://docs.apollo.io/reference/get-a-list-of-email-accounts\" target=\"_blank\">Get a List of Email Accounts endpoint</a> and identify the `id` value for the email account. <br><br>Example: `6633baaece5fbd01c791d7ca`",
                type: "Query",
                schema: addContactsToSequenceQueryParamsSchema.shape["send_email_from_email_account_id"]
            },
            {
                name: "sequence_no_email",
                description: "Set to `true` if you want to add contacts to the sequence even if they do not have an email address.",
                type: "Query",
                schema: addContactsToSequenceQueryParamsSchema.shape["sequence_no_email"]
            },
            {
                name: "sequence_unverified_email",
                description: "Set to `true` if you want to add contacts to the sequence if they have an unverified email address.",
                type: "Query",
                schema: addContactsToSequenceQueryParamsSchema.shape["sequence_unverified_email"]
            },
            {
                name: "sequence_job_change",
                description: "Set to `true` if you want to add contacts to the sequence even if they have recently changed jobs.",
                type: "Query",
                schema: addContactsToSequenceQueryParamsSchema.shape["sequence_job_change"]
            },
            {
                name: "sequence_active_in_other_campaigns",
                description: "Set to `true` if you want to add contacts to the sequence even if they have been added to other sequences. This parameter does not differentiate between active and paused sequences.",
                type: "Query",
                schema: addContactsToSequenceQueryParamsSchema.shape["sequence_active_in_other_campaigns"]
            },
            {
                name: "sequence_finished_in_other_campaigns",
                description: "Set to `true` if you want to add contacts to the sequence if they have been marked as `finished` in another sequence.",
                type: "Query",
                schema: addContactsToSequenceQueryParamsSchema.shape["sequence_finished_in_other_campaigns"]
            },
            {
                name: "user_id",
                description: "The ID for the user in your team's Apollo account. <br><br>This is the user taking the action to add contacts to a sequence. When the sequence is updated, the activity log shows the user that added the contacts. <br><br>Use the <a href=\"https://docs.apollo.io/reference/get-a-list-of-users\" target=\"_blank\">Get a List of Users endpoint</a> to retrieve IDs for all of the users within your Apollo account. <br><br>Example: `66302798d03b9601c7934ebf`",
                type: "Query",
                schema: addContactsToSequenceQueryParamsSchema.shape["user_id"]
            }
        ],
        response: addContactsToSequenceMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: addContactsToSequence401Schema
            },
            {
                status: 403,
                description: "403",
                schema: addContactsToSequence403Schema
            },
            {
                status: 422,
                description: "422",
                schema: addContactsToSequence422Schema
            },
            {
                status: 429,
                description: "429",
                schema: addContactsToSequence429Schema
            }
        ],
        alias: "addContactIdsToEmailerCampaign",
    },
    {
        method: "post",
        path: "/emailer_campaigns/remove_or_stop_contact_ids",
        description: "Use the Update Contact Status in a Sequence endpoint to either mark contacts as having `finished` a sequence, or to remove them from a sequence entirely. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [
            {
                name: "emailer_campaign_ids[]",
                description: "The Apollo IDs for the sequences that you want to update. If you add multiple sequences, you will update the status of the contacts across the chosen sequences. <br><br>To find sequence IDs, call the <a href=\"https://docs.apollo.io/reference/search-for-sequences\" target=\"_blank\">Search for Sequences endpoint</a> and identify the `id` value for the sequence. <br><br>Example: `66e9e215ece19801b219997f`",
                type: "Query",
                schema: updateContactStatusSequenceQueryParamsSchema.shape["emailer_campaign_ids[]"]
            },
            {
                name: "contact_ids[]",
                description: "The Apollo IDs for the contacts in the sequences. These are the contacts whose sequence status you want to update. <br><br>To find contact IDs, call the <a href=\"https://docs.apollo.io/reference/search-for-contacts\" target=\"_blank\">Search for Contacts endpoint</a> and identify the `id` value for the contact. <br><br>Example: `66e34b81740c50074e3d1bd4`",
                type: "Query",
                schema: updateContactStatusSequenceQueryParamsSchema.shape["contact_ids[]"]
            },
            {
                name: "mode",
                description: "Choose 1 of the following options to update the sequence status of the contacts:   <ul> <li> `mark_as_finished`: Mark the contacts as having finished the sequence. </li> <li> `remove`: Remove the contacts from the sequence. </li> <li> `stop`:  </li> </ul>",
                type: "Query",
                schema: updateContactStatusSequenceQueryParamsSchema.shape["mode"]
            }
        ],
        response: updateContactStatusSequenceMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: updateContactStatusSequence401Schema
            },
            {
                status: 403,
                description: "403",
                schema: updateContactStatusSequence403Schema
            },
            {
                status: 429,
                description: "429",
                schema: updateContactStatusSequence429Schema
            }
        ],
        alias: "updateEmailerCampaignContactStatus",
    },
    {
        method: "post",
        path: "/tasks/bulk_create",
        description: "Use the Create a Task endpoint to create tasks in Apollo for you and your team. Tasks help track the upcoming actions you need to take, such as emailing or calling a contact. <br><br>This endpoint currently only returns `true` as a response. To manually check whether a task was created appropriately, use the Apollo product. <br><br>Apollo does not apply deduplication processes when you create a new task via the API. If your entry has the same task owner, contact, and other details as an existing task, Apollo will create a new task instead of updating the existing task. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key. <br><br>This feature is not accessible to Apollo users on free plans.",
        requestFormat: "json",
        parameters: [
            {
                name: "user_id",
                description: "The ID for the task owner within your team's Apollo account. This is the user that will take action on the contacts. <br><br>Use the <a href=\"https://docs.apollo.io/reference/get-a-list-of-users\" target=\"_blank\">Get a List of Users endpoint</a> to retrieve IDs for all of the users within your Apollo account. <br><br>Example: `66302798d03b9601c7934ebf`",
                type: "Query",
                schema: createTaskQueryParamsSchema.shape["user_id"]
            },
            {
                name: "contact_ids[]",
                description: "The Apollo IDs for the contacts that you want to be on the receiving end of the action. <br><br>If you add multiple contact IDs, individual tasks will be created for each of the contacts using the same task type, due date, and other details. <br><br>To find contact IDs, call the <a href=\"https://docs.apollo.io/reference/search-for-contacts\" target=\"_blank\">Search for Contacts endpoint</a> and identify the `id` value for the contact. <br><br>Example: `66e34b81740c50074e3d1bd4`",
                type: "Query",
                schema: createTaskQueryParamsSchema.shape["contact_ids[]"]
            },
            {
                name: "priority",
                description: "Assign a priority to the task you are creating:  <ul> <li> `high` </li> <li> `medium` </li> <li> `low` </li> </ul>",
                type: "Query",
                schema: createTaskQueryParamsSchema.shape["priority"]
            },
            {
                name: "due_at",
                description: "The full date and time when the task will be due. <br><br>Your entry should adhere to the <a href=\"https://www.rfc-editor.org/rfc/rfc3339#section-5.6\" target=\"_blank\">ISO 8601 date-time format</a>. Apollo uses Greenwich Mean Time (GMT) by default. If you do not account for time zone differences, you could add a task due date that falls on a different day than you intended. <br><br>The value you enter can either adhere to GMT, or you can adjust the time manually by specifiying in hours and minutes how much you want to offset GMT.<br><br>Example: `2025-02-15T08:10:30Z`; `2025-03-25T10:15:30+05:00Z`",
                type: "Query",
                schema: createTaskQueryParamsSchema.shape["due_at"]
            },
            {
                name: "type",
                description: "Set the task to be 1 of the following task types. This enables the task owner to know the type of action they need to take. <ul> <li> `call`: Call the contacts. </li> <li> `outreach_manual_email`: Email the contacts. </li> <li> `linkedin_step_connect`: Send a LinkedIn invitation to connect with the contacts. </li> <li> `linkedin_step_message`: Send a direct message to the contacts' LinkedIn profiles. </li> <li> `linkedin_step_view_profile`: View the contacts' LinkedIn profiles. </li> <li> `linkedin_step_interact_post`: Interact with the contacts' recent LinkedIn posts. </li> <li> `action_item`: Take generic action for the contacts. If you use this task type, Apollo recommends using the `note` parameter too. </li> </ul>",
                type: "Query",
                schema: createTaskQueryParamsSchema.shape["type"]
            },
            {
                name: "status",
                description: "The status of the task being created. For future-facing tasks, you should use the `scheduled` status. <br><br>For tasks that are already completed, you can use `completed` or `archived`. <br><br>Example: `scheduled`",
                type: "Query",
                schema: createTaskQueryParamsSchema.shape["status"]
            },
            {
                name: "note",
                description: "Add a description for the task. This should be a human-readable message. <br><br>This parameter is not required, but it is recommended as it provides the task owner with more context on the action they need to take. <br><br>Example: `This contact expressed interest in the Sequences feature specifically. Be prepared to discuss.`",
                type: "Query",
                schema: createTaskQueryParamsSchema.shape["note"]
            }
        ],
        response: createTaskMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: createTask401Schema
            },
            {
                status: 403,
                description: "403",
                schema: createTask403Schema
            },
            {
                status: 429,
                description: "429",
                schema: createTask429Schema
            }
        ],
        alias: "createTask",
    },
    {
        method: "post",
        path: "/tasks/search",
        description: "Use the Search for Tasks endpoint to find tasks that your team has created in Apollo. <br><br>To protect Apollo's performance for all users, this endpoint has a display limit of 50,000 records (100 records per page, up to 500 pages). Add more filters to narrow your search results as much as possible. This limitation does not restrict your access to Apollo's database; you just need to access the data in batches. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key. <br><br>This feature is not accessible to Apollo users on free plans.",
        requestFormat: "json",
        parameters: [
            {
                name: "sort_by_field",
                description: "Sort the tasks by 1 of the following options:  <ul> <li> `task_due_at`: The most future-dated first. </li> <li> `task_priority`: The highest priority first. </li> </ul>",
                type: "Query",
                schema: searchTasksQueryParamsSchema.shape["sort_by_field"]
            },
            {
                name: "open_factor_names[]",
                description: "Enter `task_types` for this parameter to return a count of tasks by task type. <br><br>When used, the response includes a `\"task_types\": []` array with a `\"count\"` value for each task type.",
                type: "Query",
                schema: searchTasksQueryParamsSchema.shape["open_factor_names[]"]
            },
            {
                name: "page",
                description: "The page number of the Apollo data that you want to retrieve. <br><br>Use this parameter in combination with the `per_page` parameter to make search results for navigable and improve the performance of the endpoint. <br><br>Example: `4`",
                type: "Query",
                schema: searchTasksQueryParamsSchema.shape["page"]
            },
            {
                name: "per_page",
                description: "The number of search results that should be returned for each page. Limited the number of results per page improves the endpoint's performance. <br><br>Use the `page` parameter to search the different pages of data. <br><br>Example: `10`",
                type: "Query",
                schema: searchTasksQueryParamsSchema.shape["per_page"]
            }
        ],
        response: searchTasksMutationResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: searchTasks401Schema
            },
            {
                status: 403,
                description: "403",
                schema: searchTasks403Schema
            },
            {
                status: 422,
                description: "422",
                schema: searchTasks422Schema
            },
            {
                status: 429,
                description: "429",
                schema: searchTasks429Schema
            }
        ],
        alias: "searchTasks",
    },
    {
        method: "get",
        path: "/users/search",
        description: "Use the Get a List of Users endpoint to retrieve the IDs for all of the users (teammates) in your Apollo account. <br><br>These IDs can be used for several other endpoints, including the <a href=\"https://docs.apollo.io/reference/create-deal\" target=\"_blank\">Create a Deal</a>, <a href=\"https://docs.apollo.io/reference/create-an-account\" target=\"_blank\">Create an Account</a>, and <a href=\"https://docs.apollo.io/reference/create-task\" target=\"_blank\">Create a Task</a> endpoints. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key. <br><br>This feature is not accessible to Apollo users on free plans.",
        requestFormat: "json",
        parameters: [
            {
                name: "page",
                description: "The page number of the Apollo data that you want to retrieve. <br><br>Use this parameter in combination with the `per_page` parameter to make search results for navigable and improve the performance of the endpoint. <br><br>Example: `4`",
                type: "Query",
                schema: getAListOfUsersQueryParamsSchema.shape["page"]
            },
            {
                name: "per_page",
                description: "The number of search results that should be returned for each page. Limited the number of results per page improves the endpoint's performance. <br><br>Use the `page` parameter to search the different pages of data. <br><br>Example: `10`",
                type: "Query",
                schema: getAListOfUsersQueryParamsSchema.shape["per_page"]
            }
        ],
        response: getAListOfUsersQueryResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: getAListOfUsers401Schema
            },
            {
                status: 403,
                description: "403",
                schema: getAListOfUsers403Schema
            },
            {
                status: 429,
                description: "429",
                schema: getAListOfUsers429Schema
            }
        ],
        alias: "searchUsers",
    },
    {
        method: "get",
        path: "/email_accounts",
        description: "Use the Get a List of Email Accounts endpoint to retrieve information about the linked email inboxes that your teammates use in your Apollo account. <br><br>In particular, this endpoint returns IDs for each of your team's linked email accounts, which can be used with the <a href=\"https://docs.apollo.io/reference/add-contacts-to-sequence\" target=\"_blank\">Add Contacts to a Sequence endpoint</a>. <br><br>This endpoint does not require any parameters. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [],
        response: getAListOfEmailAccountsQueryResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: getAListOfEmailAccounts401Schema
            },
            {
                status: 403,
                description: "403",
                schema: getAListOfEmailAccounts403Schema
            },
            {
                status: 429,
                description: "429",
                schema: getAListOfEmailAccounts429Schema
            }
        ],
        alias: "listEmailAccounts",
    },
    {
        method: "get",
        path: "/labels",
        description: "Use the Get a List of All Lists and Tags endpoint to retrieve information about every list that has been created in your Apollo account. This endpoint can be used to check the available lists before you use the <a href=\"https://docs.apollo.io/reference/create-a-contact\" target=\"_blank\">Create a Contact endpoint</a>. <br><br>This endpoint does not require any parameters. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [],
        response: getAListOfAllListstagsQueryResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: getAListOfAllListstags401Schema
            },
            {
                status: 403,
                description: "403",
                schema: getAListOfAllListstags403Schema
            },
            {
                status: 429,
                description: "429",
                schema: getAListOfAllListstags429Schema
            }
        ],
        alias: "listTagsAndLists",
    },
    {
        method: "get",
        path: "/typed_custom_fields",
        description: "Use the Get a List of Custom Fields endpoint to retrieve information about all of the custom fields that have been created in your Apollo account. <br><br>This endpoint does not require any parameters. <br><br>This endpoint requires a master API key. If you attempt to call the endpoint without a master key, you will receive a `403` response. Refer to <a href=\"https://docs.apollo.io/docs/create-api-key\" target=\"_blank\">Create API Keys</a> to learn how to create a master API key.",
        requestFormat: "json",
        parameters: [],
        response: getAListOfAllCustomFieldsQueryResponseSchema,
        errors: [
            {
                status: 401,
                description: "401",
                schema: getAListOfAllCustomFields401Schema
            },
            {
                status: 403,
                description: "403",
                schema: getAListOfAllCustomFields403Schema
            },
            {
                status: 429,
                description: "429",
                schema: getAListOfAllCustomFields429Schema
            }
        ],
        alias: "listCustomFields",
    }
]);
export var getAPI = function (baseUrl) { return new Zodios(baseUrl, endpoints); };
export var api = new Zodios("https://api.apollo.io/api/v1", endpoints);
export default api;
