// @hubspot/api-client/lib/codegen/automation/actions/models/ObjectSerializer.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectSerializer = void 0;
__exportStar(require("../models/ActionFunction"), exports);
__exportStar(require("../models/ActionFunctionIdentifier"), exports);
__exportStar(require("../models/ActionLabels"), exports);
__exportStar(require("../models/ActionRevision"), exports);
__exportStar(require("../models/BatchInputCallbackCompletionBatchRequest"), exports);
__exportStar(require("../models/CallbackCompletionBatchRequest"), exports);
__exportStar(require("../models/CallbackCompletionRequest"), exports);
__exportStar(require("../models/CollectionResponseActionFunctionIdentifierNoPaging"), exports);
__exportStar(require("../models/CollectionResponseActionRevisionForwardPaging"), exports);
__exportStar(require("../models/CollectionResponseExtensionActionDefinitionForwardPaging"), exports);
__exportStar(require("../models/ConditionalSingleFieldDependency"), exports);
__exportStar(require("../models/ErrorDetail"), exports);
__exportStar(require("../models/ExtensionActionDefinition"), exports);
__exportStar(require("../models/ExtensionActionDefinitionInput"), exports);
__exportStar(require("../models/ExtensionActionDefinitionPatch"), exports);
__exportStar(require("../models/ExtensionActionDefinitionPatchInputFieldDependenciesInner"), exports);
__exportStar(require("../models/FieldTypeDefinition"), exports);
__exportStar(require("../models/ForwardPaging"), exports);
__exportStar(require("../models/InputFieldDefinition"), exports);
__exportStar(require("../models/ModelError"), exports);
__exportStar(require("../models/NextPage"), exports);
__exportStar(require("../models/ObjectRequestOptions"), exports);
__exportStar(require("../models/Option"), exports);
__exportStar(require("../models/SingleFieldDependency"), exports);
const ActionFunction_1 = require("../models/ActionFunction");
const ActionFunctionIdentifier_1 = require("../models/ActionFunctionIdentifier");
const ActionLabels_1 = require("../models/ActionLabels");
const ActionRevision_1 = require("../models/ActionRevision");
const BatchInputCallbackCompletionBatchRequest_1 = require("../models/BatchInputCallbackCompletionBatchRequest");
const CallbackCompletionBatchRequest_1 = require("../models/CallbackCompletionBatchRequest");
const CallbackCompletionRequest_1 = require("../models/CallbackCompletionRequest");
const CollectionResponseActionFunctionIdentifierNoPaging_1 = require("../models/CollectionResponseActionFunctionIdentifierNoPaging");
const CollectionResponseActionRevisionForwardPaging_1 = require("../models/CollectionResponseActionRevisionForwardPaging");
const CollectionResponseExtensionActionDefinitionForwardPaging_1 = require("../models/CollectionResponseExtensionActionDefinitionForwardPaging");
const ConditionalSingleFieldDependency_1 = require("../models/ConditionalSingleFieldDependency");
const ErrorDetail_1 = require("../models/ErrorDetail");
const ExtensionActionDefinition_1 = require("../models/ExtensionActionDefinition");
const ExtensionActionDefinitionInput_1 = require("../models/ExtensionActionDefinitionInput");
const ExtensionActionDefinitionPatch_1 = require("../models/ExtensionActionDefinitionPatch");
const ExtensionActionDefinitionPatchInputFieldDependenciesInner_1 = require("../models/ExtensionActionDefinitionPatchInputFieldDependenciesInner");
const FieldTypeDefinition_1 = require("../models/FieldTypeDefinition");
const ForwardPaging_1 = require("../models/ForwardPaging");
const InputFieldDefinition_1 = require("../models/InputFieldDefinition");
const ModelError_1 = require("../models/ModelError");
const NextPage_1 = require("../models/NextPage");
const ObjectRequestOptions_1 = require("../models/ObjectRequestOptions");
const Option_1 = require("../models/Option");
const SingleFieldDependency_1 = require("../models/SingleFieldDependency");
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
const supportedMediaTypes = {
    "application/json": Infinity,
    "application/octet-stream": 0,
    "application/x-www-form-urlencoded": 0,
    "text/plain": 0
};
let enumsMap = new Set([
    "ActionFunctionFunctionTypeEnum",
    "ActionFunctionIdentifierFunctionTypeEnum",
    "ConditionalSingleFieldDependencyDependencyTypeEnum",
    "ExtensionActionDefinitionPatchInputFieldDependenciesInnerDependencyTypeEnum",
    "FieldTypeDefinitionTypeEnum",
    "FieldTypeDefinitionFieldTypeEnum",
    "FieldTypeDefinitionReferencedObjectTypeEnum",
    "InputFieldDefinitionSupportedValueTypesEnum",
    "SingleFieldDependencyDependencyTypeEnum",
]);
let typeMap = {
    "ActionFunction": ActionFunction_1.ActionFunction,
    "ActionFunctionIdentifier": ActionFunctionIdentifier_1.ActionFunctionIdentifier,
    "ActionLabels": ActionLabels_1.ActionLabels,
    "ActionRevision": ActionRevision_1.ActionRevision,
    "BatchInputCallbackCompletionBatchRequest": BatchInputCallbackCompletionBatchRequest_1.BatchInputCallbackCompletionBatchRequest,
    "CallbackCompletionBatchRequest": CallbackCompletionBatchRequest_1.CallbackCompletionBatchRequest,
    "CallbackCompletionRequest": CallbackCompletionRequest_1.CallbackCompletionRequest,
    "CollectionResponseActionFunctionIdentifierNoPaging": CollectionResponseActionFunctionIdentifierNoPaging_1.CollectionResponseActionFunctionIdentifierNoPaging,
    "CollectionResponseActionRevisionForwardPaging": CollectionResponseActionRevisionForwardPaging_1.CollectionResponseActionRevisionForwardPaging,
    "CollectionResponseExtensionActionDefinitionForwardPaging": CollectionResponseExtensionActionDefinitionForwardPaging_1.CollectionResponseExtensionActionDefinitionForwardPaging,
    "ConditionalSingleFieldDependency": ConditionalSingleFieldDependency_1.ConditionalSingleFieldDependency,
    "ErrorDetail": ErrorDetail_1.ErrorDetail,
    "ExtensionActionDefinition": ExtensionActionDefinition_1.ExtensionActionDefinition,
    "ExtensionActionDefinitionInput": ExtensionActionDefinitionInput_1.ExtensionActionDefinitionInput,
    "ExtensionActionDefinitionPatch": ExtensionActionDefinitionPatch_1.ExtensionActionDefinitionPatch,
    "ExtensionActionDefinitionPatchInputFieldDependenciesInner": ExtensionActionDefinitionPatchInputFieldDependenciesInner_1.ExtensionActionDefinitionPatchInputFieldDependenciesInner,
    "FieldTypeDefinition": FieldTypeDefinition_1.FieldTypeDefinition,
    "ForwardPaging": ForwardPaging_1.ForwardPaging,
    "InputFieldDefinition": InputFieldDefinition_1.InputFieldDefinition,
    "ModelError": ModelError_1.ModelError,
    "NextPage": NextPage_1.NextPage,
    "ObjectRequestOptions": ObjectRequestOptions_1.ObjectRequestOptions,
    "Option": Option_1.Option,
    "SingleFieldDependency": SingleFieldDependency_1.SingleFieldDependency,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap.has(expectedType)) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType;
            }
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType;
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType;
                    }
                    else {
                        return expectedType;
                    }
                }
                else {
                    return expectedType;
                }
            }
        }
    }
    static serialize(data, type, format) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) {
            let subType = type.replace("Array<", "");
            subType = subType.substring(0, subType.length - 1);
            let transformedData = [];
            for (let date of data) {
                transformedData.push(ObjectSerializer.serialize(date, subType, format));
            }
            return transformedData;
        }
        else if (type === "Date") {
            if (format == "date") {
                let month = data.getMonth() + 1;
                month = month < 10 ? "0" + month.toString() : month.toString();
                let day = data.getDate();
                day = day < 10 ? "0" + day.toString() : day.toString();
                return data.getFullYear() + "-" + month + "-" + day;
            }
            else {
                return data.toISOString();
            }
        }
        else {
            if (enumsMap.has(type)) {
                return data;
            }
            if (!typeMap[type]) {
                return data;
            }
            type = this.findCorrectType(data, type);
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let attributeType of attributeTypes) {
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type, attributeType.format);
            }
            return instance;
        }
    }
    static deserialize(data, type, format) {
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) {
            let subType = type.replace("Array<", "");
            subType = subType.substring(0, subType.length - 1);
            let transformedData = [];
            for (let date of data) {
                transformedData.push(ObjectSerializer.deserialize(date, subType, format));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap.has(type)) {
                return data;
            }
            if (!typeMap[type]) {
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let attributeType of attributeTypes) {
                let value = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type, attributeType.format);
                if (value !== undefined) {
                    instance[attributeType.name] = value;
                }
            }
            return instance;
        }
    }
    static normalizeMediaType(mediaType) {
        if (mediaType === undefined) {
            return undefined;
        }
        return mediaType.split(";")[0].trim().toLowerCase();
    }
    static getPreferredMediaType(mediaTypes) {
        if (!mediaTypes) {
            return "application/json";
        }
        const normalMediaTypes = mediaTypes.map(this.normalizeMediaType);
        let selectedMediaType = undefined;
        let selectedRank = -Infinity;
        for (const mediaType of normalMediaTypes) {
            if (supportedMediaTypes[mediaType] > selectedRank) {
                selectedMediaType = mediaType;
                selectedRank = supportedMediaTypes[mediaType];
            }
        }
        if (selectedMediaType === undefined) {
            throw new Error("None of the given media types are supported: " + mediaTypes.join(", "));
        }
        return selectedMediaType;
    }
    static stringify(data, mediaType) {
        if (mediaType === "text/plain") {
            return String(data);
        }
        if (mediaType === "application/json") {
            return JSON.stringify(data);
        }
        throw new Error("The mediaType " + mediaType + " is not supported by ObjectSerializer.stringify.");
    }
    static parse(rawData, mediaType) {
        if (mediaType === undefined) {
            throw new Error("Cannot parse content. No Content-Type defined.");
        }
        if (mediaType === "text/plain") {
            return rawData;
        }
        if (mediaType === "application/json") {
            return JSON.parse(rawData);
        }
        if (mediaType === "text/html") {
            return rawData;
        }
        throw new Error("The mediaType " + mediaType + " is not supported by ObjectSerializer.parse.");
    }
}
exports.ObjectSerializer = ObjectSerializer;
