// @hubspot/api-client/lib/codegen/cms/pages/types/ObjectParamAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectSitePagesApi = exports.ObjectLandingPagesApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class ObjectLandingPagesApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableLandingPagesApi(configuration, requestFactory, responseProcessor);
    }
    archive(param, options) {
        return this.api.archive(param.objectId, param.archived, options).toPromise();
    }
    archiveBatch(param, options) {
        return this.api.archiveBatch(param.batchInputString, options).toPromise();
    }
    archiveFolder(param, options) {
        return this.api.archiveFolder(param.objectId, param.archived, options).toPromise();
    }
    archiveFolders(param, options) {
        return this.api.archiveFolders(param.batchInputString, options).toPromise();
    }
    attachToLangGroup(param, options) {
        return this.api.attachToLangGroup(param.attachToLangPrimaryRequestVNext, options).toPromise();
    }
    clone(param, options) {
        return this.api.clone(param.contentCloneRequestVNext, options).toPromise();
    }
    create(param, options) {
        return this.api.create(param.page, options).toPromise();
    }
    createABTestVariation(param, options) {
        return this.api.createABTestVariation(param.abTestCreateRequestVNext, options).toPromise();
    }
    createBatch(param, options) {
        return this.api.createBatch(param.batchInputPage, options).toPromise();
    }
    createFolder(param, options) {
        return this.api.createFolder(param.contentFolder, options).toPromise();
    }
    createFolders(param, options) {
        return this.api.createFolders(param.batchInputContentFolder, options).toPromise();
    }
    createLangVariation(param, options) {
        return this.api.createLangVariation(param.contentLanguageCloneRequestVNext, options).toPromise();
    }
    detachFromLangGroup(param, options) {
        return this.api.detachFromLangGroup(param.detachFromLangGroupRequestVNext, options).toPromise();
    }
    endActiveABTest(param, options) {
        return this.api.endActiveABTest(param.abTestEndRequestVNext, options).toPromise();
    }
    getById(param, options) {
        return this.api.getById(param.objectId, param.archived, param.property, options).toPromise();
    }
    getDraftById(param, options) {
        return this.api.getDraftById(param.objectId, options).toPromise();
    }
    getFolderById(param, options) {
        return this.api.getFolderById(param.objectId, param.archived, param.property, options).toPromise();
    }
    getFolderPreviousVersion(param, options) {
        return this.api.getFolderPreviousVersion(param.objectId, param.revisionId, options).toPromise();
    }
    getFolderPreviousVersions(param, options) {
        return this.api.getFolderPreviousVersions(param.objectId, param.after, param.before, param.limit, options).toPromise();
    }
    getFoldersPage(param = {}, options) {
        return this.api.getFoldersPage(param.createdAt, param.createdAfter, param.createdBefore, param.updatedAt, param.updatedAfter, param.updatedBefore, param.sort, param.after, param.limit, param.archived, param.property, options).toPromise();
    }
    getPage(param = {}, options) {
        return this.api.getPage(param.createdAt, param.createdAfter, param.createdBefore, param.updatedAt, param.updatedAfter, param.updatedBefore, param.sort, param.after, param.limit, param.archived, param.property, options).toPromise();
    }
    getPreviousVersion(param, options) {
        return this.api.getPreviousVersion(param.objectId, param.revisionId, options).toPromise();
    }
    getPreviousVersions(param, options) {
        return this.api.getPreviousVersions(param.objectId, param.after, param.before, param.limit, options).toPromise();
    }
    pushLive(param, options) {
        return this.api.pushLive(param.objectId, options).toPromise();
    }
    readBatch(param, options) {
        return this.api.readBatch(param.batchInputString, param.archived, options).toPromise();
    }
    readFolders(param, options) {
        return this.api.readFolders(param.batchInputString, param.archived, options).toPromise();
    }
    rerunPreviousABTest(param, options) {
        return this.api.rerunPreviousABTest(param.abTestRerunRequestVNext, options).toPromise();
    }
    resetDraft(param, options) {
        return this.api.resetDraft(param.objectId, options).toPromise();
    }
    restoreFolderPreviousVersion(param, options) {
        return this.api.restoreFolderPreviousVersion(param.objectId, param.revisionId, options).toPromise();
    }
    restorePreviousVersion(param, options) {
        return this.api.restorePreviousVersion(param.objectId, param.revisionId, options).toPromise();
    }
    restorePreviousVersionToDraft(param, options) {
        return this.api.restorePreviousVersionToDraft(param.objectId, param.revisionId, options).toPromise();
    }
    schedule(param, options) {
        return this.api.schedule(param.contentScheduleRequestVNext, options).toPromise();
    }
    setLangPrimary(param, options) {
        return this.api.setLangPrimary(param.setNewLanguagePrimaryRequestVNext, options).toPromise();
    }
    update(param, options) {
        return this.api.update(param.objectId, param.page, param.archived, options).toPromise();
    }
    updateBatch(param, options) {
        return this.api.updateBatch(param.batchInputJsonNode, param.archived, options).toPromise();
    }
    updateDraft(param, options) {
        return this.api.updateDraft(param.objectId, param.page, options).toPromise();
    }
    updateFolder(param, options) {
        return this.api.updateFolder(param.objectId, param.contentFolder, param.archived, options).toPromise();
    }
    updateFolders(param, options) {
        return this.api.updateFolders(param.batchInputJsonNode, param.archived, options).toPromise();
    }
    updateLangs(param, options) {
        return this.api.updateLangs(param.updateLanguagesRequestVNext, options).toPromise();
    }
}
exports.ObjectLandingPagesApi = ObjectLandingPagesApi;
const ObservableAPI_2 = require("./ObservableAPI");
class ObjectSitePagesApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableSitePagesApi(configuration, requestFactory, responseProcessor);
    }
    archive(param, options) {
        return this.api.archive(param.objectId, param.archived, options).toPromise();
    }
    archiveBatch(param, options) {
        return this.api.archiveBatch(param.batchInputString, options).toPromise();
    }
    attachToLangGroup(param, options) {
        return this.api.attachToLangGroup(param.attachToLangPrimaryRequestVNext, options).toPromise();
    }
    clone(param, options) {
        return this.api.clone(param.contentCloneRequestVNext, options).toPromise();
    }
    create(param, options) {
        return this.api.create(param.page, options).toPromise();
    }
    createABTestVariation(param, options) {
        return this.api.createABTestVariation(param.abTestCreateRequestVNext, options).toPromise();
    }
    createBatch(param, options) {
        return this.api.createBatch(param.batchInputPage, options).toPromise();
    }
    createLangVariation(param, options) {
        return this.api.createLangVariation(param.contentLanguageCloneRequestVNext, options).toPromise();
    }
    detachFromLangGroup(param, options) {
        return this.api.detachFromLangGroup(param.detachFromLangGroupRequestVNext, options).toPromise();
    }
    endActiveABTest(param, options) {
        return this.api.endActiveABTest(param.abTestEndRequestVNext, options).toPromise();
    }
    getById(param, options) {
        return this.api.getById(param.objectId, param.archived, param.property, options).toPromise();
    }
    getDraftById(param, options) {
        return this.api.getDraftById(param.objectId, options).toPromise();
    }
    getPage(param = {}, options) {
        return this.api.getPage(param.createdAt, param.createdAfter, param.createdBefore, param.updatedAt, param.updatedAfter, param.updatedBefore, param.sort, param.after, param.limit, param.archived, param.property, options).toPromise();
    }
    getPreviousVersion(param, options) {
        return this.api.getPreviousVersion(param.objectId, param.revisionId, options).toPromise();
    }
    getPreviousVersions(param, options) {
        return this.api.getPreviousVersions(param.objectId, param.after, param.before, param.limit, options).toPromise();
    }
    pushLive(param, options) {
        return this.api.pushLive(param.objectId, options).toPromise();
    }
    readBatch(param, options) {
        return this.api.readBatch(param.batchInputString, param.archived, options).toPromise();
    }
    rerunPreviousABTest(param, options) {
        return this.api.rerunPreviousABTest(param.abTestRerunRequestVNext, options).toPromise();
    }
    resetDraft(param, options) {
        return this.api.resetDraft(param.objectId, options).toPromise();
    }
    restorePreviousVersion(param, options) {
        return this.api.restorePreviousVersion(param.objectId, param.revisionId, options).toPromise();
    }
    restorePreviousVersionToDraft(param, options) {
        return this.api.restorePreviousVersionToDraft(param.objectId, param.revisionId, options).toPromise();
    }
    schedule(param, options) {
        return this.api.schedule(param.contentScheduleRequestVNext, options).toPromise();
    }
    setLangPrimary(param, options) {
        return this.api.setLangPrimary(param.setNewLanguagePrimaryRequestVNext, options).toPromise();
    }
    update(param, options) {
        return this.api.update(param.objectId, param.page, param.archived, options).toPromise();
    }
    updateBatch(param, options) {
        return this.api.updateBatch(param.batchInputJsonNode, param.archived, options).toPromise();
    }
    updateDraft(param, options) {
        return this.api.updateDraft(param.objectId, param.page, options).toPromise();
    }
    updateLangs(param, options) {
        return this.api.updateLangs(param.updateLanguagesRequestVNext, options).toPromise();
    }
}
exports.ObjectSitePagesApi = ObjectSitePagesApi;
