// @hubspot/api-client/lib/codegen/cms/pages/types/PromiseAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromiseSitePagesApi = exports.PromiseLandingPagesApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class PromiseLandingPagesApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableLandingPagesApi(configuration, requestFactory, responseProcessor);
    }
    archive(objectId, archived, _options) {
        const result = this.api.archive(objectId, archived, _options);
        return result.toPromise();
    }
    archiveBatch(batchInputString, _options) {
        const result = this.api.archiveBatch(batchInputString, _options);
        return result.toPromise();
    }
    archiveFolder(objectId, archived, _options) {
        const result = this.api.archiveFolder(objectId, archived, _options);
        return result.toPromise();
    }
    archiveFolders(batchInputString, _options) {
        const result = this.api.archiveFolders(batchInputString, _options);
        return result.toPromise();
    }
    attachToLangGroup(attachToLangPrimaryRequestVNext, _options) {
        const result = this.api.attachToLangGroup(attachToLangPrimaryRequestVNext, _options);
        return result.toPromise();
    }
    clone(contentCloneRequestVNext, _options) {
        const result = this.api.clone(contentCloneRequestVNext, _options);
        return result.toPromise();
    }
    create(page, _options) {
        const result = this.api.create(page, _options);
        return result.toPromise();
    }
    createABTestVariation(abTestCreateRequestVNext, _options) {
        const result = this.api.createABTestVariation(abTestCreateRequestVNext, _options);
        return result.toPromise();
    }
    createBatch(batchInputPage, _options) {
        const result = this.api.createBatch(batchInputPage, _options);
        return result.toPromise();
    }
    createFolder(contentFolder, _options) {
        const result = this.api.createFolder(contentFolder, _options);
        return result.toPromise();
    }
    createFolders(batchInputContentFolder, _options) {
        const result = this.api.createFolders(batchInputContentFolder, _options);
        return result.toPromise();
    }
    createLangVariation(contentLanguageCloneRequestVNext, _options) {
        const result = this.api.createLangVariation(contentLanguageCloneRequestVNext, _options);
        return result.toPromise();
    }
    detachFromLangGroup(detachFromLangGroupRequestVNext, _options) {
        const result = this.api.detachFromLangGroup(detachFromLangGroupRequestVNext, _options);
        return result.toPromise();
    }
    endActiveABTest(abTestEndRequestVNext, _options) {
        const result = this.api.endActiveABTest(abTestEndRequestVNext, _options);
        return result.toPromise();
    }
    getById(objectId, archived, property, _options) {
        const result = this.api.getById(objectId, archived, property, _options);
        return result.toPromise();
    }
    getDraftById(objectId, _options) {
        const result = this.api.getDraftById(objectId, _options);
        return result.toPromise();
    }
    getFolderById(objectId, archived, property, _options) {
        const result = this.api.getFolderById(objectId, archived, property, _options);
        return result.toPromise();
    }
    getFolderPreviousVersion(objectId, revisionId, _options) {
        const result = this.api.getFolderPreviousVersion(objectId, revisionId, _options);
        return result.toPromise();
    }
    getFolderPreviousVersions(objectId, after, before, limit, _options) {
        const result = this.api.getFolderPreviousVersions(objectId, after, before, limit, _options);
        return result.toPromise();
    }
    getFoldersPage(createdAt, createdAfter, createdBefore, updatedAt, updatedAfter, updatedBefore, sort, after, limit, archived, property, _options) {
        const result = this.api.getFoldersPage(createdAt, createdAfter, createdBefore, updatedAt, updatedAfter, updatedBefore, sort, after, limit, archived, property, _options);
        return result.toPromise();
    }
    getPage(createdAt, createdAfter, createdBefore, updatedAt, updatedAfter, updatedBefore, sort, after, limit, archived, property, _options) {
        const result = this.api.getPage(createdAt, createdAfter, createdBefore, updatedAt, updatedAfter, updatedBefore, sort, after, limit, archived, property, _options);
        return result.toPromise();
    }
    getPreviousVersion(objectId, revisionId, _options) {
        const result = this.api.getPreviousVersion(objectId, revisionId, _options);
        return result.toPromise();
    }
    getPreviousVersions(objectId, after, before, limit, _options) {
        const result = this.api.getPreviousVersions(objectId, after, before, limit, _options);
        return result.toPromise();
    }
    pushLive(objectId, _options) {
        const result = this.api.pushLive(objectId, _options);
        return result.toPromise();
    }
    readBatch(batchInputString, archived, _options) {
        const result = this.api.readBatch(batchInputString, archived, _options);
        return result.toPromise();
    }
    readFolders(batchInputString, archived, _options) {
        const result = this.api.readFolders(batchInputString, archived, _options);
        return result.toPromise();
    }
    rerunPreviousABTest(abTestRerunRequestVNext, _options) {
        const result = this.api.rerunPreviousABTest(abTestRerunRequestVNext, _options);
        return result.toPromise();
    }
    resetDraft(objectId, _options) {
        const result = this.api.resetDraft(objectId, _options);
        return result.toPromise();
    }
    restoreFolderPreviousVersion(objectId, revisionId, _options) {
        const result = this.api.restoreFolderPreviousVersion(objectId, revisionId, _options);
        return result.toPromise();
    }
    restorePreviousVersion(objectId, revisionId, _options) {
        const result = this.api.restorePreviousVersion(objectId, revisionId, _options);
        return result.toPromise();
    }
    restorePreviousVersionToDraft(objectId, revisionId, _options) {
        const result = this.api.restorePreviousVersionToDraft(objectId, revisionId, _options);
        return result.toPromise();
    }
    schedule(contentScheduleRequestVNext, _options) {
        const result = this.api.schedule(contentScheduleRequestVNext, _options);
        return result.toPromise();
    }
    setLangPrimary(setNewLanguagePrimaryRequestVNext, _options) {
        const result = this.api.setLangPrimary(setNewLanguagePrimaryRequestVNext, _options);
        return result.toPromise();
    }
    update(objectId, page, archived, _options) {
        const result = this.api.update(objectId, page, archived, _options);
        return result.toPromise();
    }
    updateBatch(batchInputJsonNode, archived, _options) {
        const result = this.api.updateBatch(batchInputJsonNode, archived, _options);
        return result.toPromise();
    }
    updateDraft(objectId, page, _options) {
        const result = this.api.updateDraft(objectId, page, _options);
        return result.toPromise();
    }
    updateFolder(objectId, contentFolder, archived, _options) {
        const result = this.api.updateFolder(objectId, contentFolder, archived, _options);
        return result.toPromise();
    }
    updateFolders(batchInputJsonNode, archived, _options) {
        const result = this.api.updateFolders(batchInputJsonNode, archived, _options);
        return result.toPromise();
    }
    updateLangs(updateLanguagesRequestVNext, _options) {
        const result = this.api.updateLangs(updateLanguagesRequestVNext, _options);
        return result.toPromise();
    }
}
exports.PromiseLandingPagesApi = PromiseLandingPagesApi;
const ObservableAPI_2 = require("./ObservableAPI");
class PromiseSitePagesApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_2.ObservableSitePagesApi(configuration, requestFactory, responseProcessor);
    }
    archive(objectId, archived, _options) {
        const result = this.api.archive(objectId, archived, _options);
        return result.toPromise();
    }
    archiveBatch(batchInputString, _options) {
        const result = this.api.archiveBatch(batchInputString, _options);
        return result.toPromise();
    }
    attachToLangGroup(attachToLangPrimaryRequestVNext, _options) {
        const result = this.api.attachToLangGroup(attachToLangPrimaryRequestVNext, _options);
        return result.toPromise();
    }
    clone(contentCloneRequestVNext, _options) {
        const result = this.api.clone(contentCloneRequestVNext, _options);
        return result.toPromise();
    }
    create(page, _options) {
        const result = this.api.create(page, _options);
        return result.toPromise();
    }
    createABTestVariation(abTestCreateRequestVNext, _options) {
        const result = this.api.createABTestVariation(abTestCreateRequestVNext, _options);
        return result.toPromise();
    }
    createBatch(batchInputPage, _options) {
        const result = this.api.createBatch(batchInputPage, _options);
        return result.toPromise();
    }
    createLangVariation(contentLanguageCloneRequestVNext, _options) {
        const result = this.api.createLangVariation(contentLanguageCloneRequestVNext, _options);
        return result.toPromise();
    }
    detachFromLangGroup(detachFromLangGroupRequestVNext, _options) {
        const result = this.api.detachFromLangGroup(detachFromLangGroupRequestVNext, _options);
        return result.toPromise();
    }
    endActiveABTest(abTestEndRequestVNext, _options) {
        const result = this.api.endActiveABTest(abTestEndRequestVNext, _options);
        return result.toPromise();
    }
    getById(objectId, archived, property, _options) {
        const result = this.api.getById(objectId, archived, property, _options);
        return result.toPromise();
    }
    getDraftById(objectId, _options) {
        const result = this.api.getDraftById(objectId, _options);
        return result.toPromise();
    }
    getPage(createdAt, createdAfter, createdBefore, updatedAt, updatedAfter, updatedBefore, sort, after, limit, archived, property, _options) {
        const result = this.api.getPage(createdAt, createdAfter, createdBefore, updatedAt, updatedAfter, updatedBefore, sort, after, limit, archived, property, _options);
        return result.toPromise();
    }
    getPreviousVersion(objectId, revisionId, _options) {
        const result = this.api.getPreviousVersion(objectId, revisionId, _options);
        return result.toPromise();
    }
    getPreviousVersions(objectId, after, before, limit, _options) {
        const result = this.api.getPreviousVersions(objectId, after, before, limit, _options);
        return result.toPromise();
    }
    pushLive(objectId, _options) {
        const result = this.api.pushLive(objectId, _options);
        return result.toPromise();
    }
    readBatch(batchInputString, archived, _options) {
        const result = this.api.readBatch(batchInputString, archived, _options);
        return result.toPromise();
    }
    rerunPreviousABTest(abTestRerunRequestVNext, _options) {
        const result = this.api.rerunPreviousABTest(abTestRerunRequestVNext, _options);
        return result.toPromise();
    }
    resetDraft(objectId, _options) {
        const result = this.api.resetDraft(objectId, _options);
        return result.toPromise();
    }
    restorePreviousVersion(objectId, revisionId, _options) {
        const result = this.api.restorePreviousVersion(objectId, revisionId, _options);
        return result.toPromise();
    }
    restorePreviousVersionToDraft(objectId, revisionId, _options) {
        const result = this.api.restorePreviousVersionToDraft(objectId, revisionId, _options);
        return result.toPromise();
    }
    schedule(contentScheduleRequestVNext, _options) {
        const result = this.api.schedule(contentScheduleRequestVNext, _options);
        return result.toPromise();
    }
    setLangPrimary(setNewLanguagePrimaryRequestVNext, _options) {
        const result = this.api.setLangPrimary(setNewLanguagePrimaryRequestVNext, _options);
        return result.toPromise();
    }
    update(objectId, page, archived, _options) {
        const result = this.api.update(objectId, page, archived, _options);
        return result.toPromise();
    }
    updateBatch(batchInputJsonNode, archived, _options) {
        const result = this.api.updateBatch(batchInputJsonNode, archived, _options);
        return result.toPromise();
    }
    updateDraft(objectId, page, _options) {
        const result = this.api.updateDraft(objectId, page, _options);
        return result.toPromise();
    }
    updateLangs(updateLanguagesRequestVNext, _options) {
        const result = this.api.updateLangs(updateLanguagesRequestVNext, _options);
        return result.toPromise();
    }
}
exports.PromiseSitePagesApi = PromiseSitePagesApi;
