// @hubspot/api-client/lib/codegen/crm/pipelines/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PipelinesApi = exports.PipelineStagesApi = exports.PipelineStageAuditsApi = exports.PipelineAuditsApi = exports.RequiredError = exports.createConfiguration = void 0;
__exportStar(require("./http/http"), exports);
__exportStar(require("./auth/auth"), exports);
__exportStar(require("./models/all"), exports);
var configuration_1 = require("./configuration");
Object.defineProperty(exports, "createConfiguration", { enumerable: true, get: function () { return configuration_1.createConfiguration; } });
__exportStar(require("./apis/exception"), exports);
__exportStar(require("./servers"), exports);
var baseapi_1 = require("./apis/baseapi");
Object.defineProperty(exports, "RequiredError", { enumerable: true, get: function () { return baseapi_1.RequiredError; } });
var PromiseAPI_1 = require("./types/PromiseAPI");
Object.defineProperty(exports, "PipelineAuditsApi", { enumerable: true, get: function () { return PromiseAPI_1.PromisePipelineAuditsApi; } });
Object.defineProperty(exports, "PipelineStageAuditsApi", { enumerable: true, get: function () { return PromiseAPI_1.PromisePipelineStageAuditsApi; } });
Object.defineProperty(exports, "PipelineStagesApi", { enumerable: true, get: function () { return PromiseAPI_1.PromisePipelineStagesApi; } });
Object.defineProperty(exports, "PipelinesApi", { enumerable: true, get: function () { return PromiseAPI_1.PromisePipelinesApi; } });
