// @hubspot/api-client/lib/src/discovery/crm/objects/taxes/TaxesDiscovery.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const configuration_1 = require("../../../../../codegen/crm/objects/taxes/configuration");
const index_1 = require("../../../../../codegen/crm/objects/taxes/index");
const rxjsStub_1 = require("../../../../../codegen/crm/objects/taxes/rxjsStub");
const ApiClientConfigurator_1 = require("../../../../configuration/ApiClientConfigurator");
const ApiDecoratorService_1 = __importDefault(require("../../../../services/ApiDecoratorService"));
class TaxesDiscovery {
    constructor(config) {
        const configuration = (0, configuration_1.createConfiguration)(ApiClientConfigurator_1.ApiClientConfigurator.getParams(config, index_1.ServerConfiguration, rxjsStub_1.Observable, rxjsStub_1.Observable));
        this.basicApi = ApiDecoratorService_1.default.getInstance().apply(new index_1.BasicApi(configuration));
        this.batchApi = ApiDecoratorService_1.default.getInstance().apply(new index_1.BatchApi(configuration));
        this.gdprApi = ApiDecoratorService_1.default.getInstance().apply(new index_1.GDPRApi(configuration));
        this.publicObjectApi = ApiDecoratorService_1.default.getInstance().apply(new index_1.PublicObjectApi(configuration));
        this.searchApi = ApiDecoratorService_1.default.getInstance().apply(new index_1.SearchApi(configuration));
    }
}
exports.default = TaxesDiscovery;
