// @hubspot/api-client/lib/codegen/marketing/forms/types/ObjectParamAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectFormsApi = void 0;
const ObservableAPI_1 = require("./ObservableAPI");
class ObjectFormsApi {
    constructor(configuration, requestFactory, responseProcessor) {
        this.api = new ObservableAPI_1.ObservableFormsApi(configuration, requestFactory, responseProcessor);
    }
    archive(param, options) {
        return this.api.archive(param.formId, options).toPromise();
    }
    create(param, options) {
        return this.api.create(param.formDefinitionCreateRequestBase, options).toPromise();
    }
    getById(param, options) {
        return this.api.getById(param.formId, param.archived, options).toPromise();
    }
    getPage(param = {}, options) {
        return this.api.getPage(param.after, param.limit, param.archived, param.formTypes, options).toPromise();
    }
    replace(param, options) {
        return this.api.replace(param.formId, param.hubSpotFormDefinition, options).toPromise();
    }
    update(param, options) {
        return this.api.update(param.formId, param.hubSpotFormDefinitionPatchRequest, options).toPromise();
    }
}
exports.ObjectFormsApi = ObjectFormsApi;
