// @hubspot/api-client/lib/codegen/crm/associations/v4/schema/models/ObjectSerializer.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectSerializer = void 0;
__exportStar(require("../models/AssociationSpec"), exports);
__exportStar(require("../models/AssociationSpecWithLabel"), exports);
__exportStar(require("../models/BatchInputPublicAssociationMultiArchive"), exports);
__exportStar(require("../models/BatchInputPublicAssociationMultiPost"), exports);
__exportStar(require("../models/BatchInputPublicDefaultAssociationMultiPost"), exports);
__exportStar(require("../models/BatchInputPublicFetchAssociationsBatchRequest"), exports);
__exportStar(require("../models/BatchResponseLabelsBetweenObjectPair"), exports);
__exportStar(require("../models/BatchResponseLabelsBetweenObjectPairWithErrors"), exports);
__exportStar(require("../models/BatchResponsePublicAssociationMultiWithLabel"), exports);
__exportStar(require("../models/BatchResponsePublicAssociationMultiWithLabelWithErrors"), exports);
__exportStar(require("../models/BatchResponsePublicDefaultAssociation"), exports);
__exportStar(require("../models/CollectionResponseAssociationSpecWithLabelNoPaging"), exports);
__exportStar(require("../models/ErrorCategory"), exports);
__exportStar(require("../models/ErrorDetail"), exports);
__exportStar(require("../models/LabelsBetweenObjectPair"), exports);
__exportStar(require("../models/ModelError"), exports);
__exportStar(require("../models/MultiAssociatedObjectWithLabel"), exports);
__exportStar(require("../models/NextPage"), exports);
__exportStar(require("../models/Paging"), exports);
__exportStar(require("../models/PreviousPage"), exports);
__exportStar(require("../models/PublicAssociationDefinitionCreateRequest"), exports);
__exportStar(require("../models/PublicAssociationDefinitionUpdateRequest"), exports);
__exportStar(require("../models/PublicAssociationMultiArchive"), exports);
__exportStar(require("../models/PublicAssociationMultiPost"), exports);
__exportStar(require("../models/PublicAssociationMultiWithLabel"), exports);
__exportStar(require("../models/PublicDefaultAssociation"), exports);
__exportStar(require("../models/PublicDefaultAssociationMultiPost"), exports);
__exportStar(require("../models/PublicFetchAssociationsBatchRequest"), exports);
__exportStar(require("../models/PublicObjectId"), exports);
__exportStar(require("../models/StandardError"), exports);
const AssociationSpec_1 = require("../models/AssociationSpec");
const AssociationSpecWithLabel_1 = require("../models/AssociationSpecWithLabel");
const BatchInputPublicAssociationMultiArchive_1 = require("../models/BatchInputPublicAssociationMultiArchive");
const BatchInputPublicAssociationMultiPost_1 = require("../models/BatchInputPublicAssociationMultiPost");
const BatchInputPublicDefaultAssociationMultiPost_1 = require("../models/BatchInputPublicDefaultAssociationMultiPost");
const BatchInputPublicFetchAssociationsBatchRequest_1 = require("../models/BatchInputPublicFetchAssociationsBatchRequest");
const BatchResponseLabelsBetweenObjectPair_1 = require("../models/BatchResponseLabelsBetweenObjectPair");
const BatchResponseLabelsBetweenObjectPairWithErrors_1 = require("../models/BatchResponseLabelsBetweenObjectPairWithErrors");
const BatchResponsePublicAssociationMultiWithLabel_1 = require("../models/BatchResponsePublicAssociationMultiWithLabel");
const BatchResponsePublicAssociationMultiWithLabelWithErrors_1 = require("../models/BatchResponsePublicAssociationMultiWithLabelWithErrors");
const BatchResponsePublicDefaultAssociation_1 = require("../models/BatchResponsePublicDefaultAssociation");
const CollectionResponseAssociationSpecWithLabelNoPaging_1 = require("../models/CollectionResponseAssociationSpecWithLabelNoPaging");
const ErrorCategory_1 = require("../models/ErrorCategory");
const ErrorDetail_1 = require("../models/ErrorDetail");
const LabelsBetweenObjectPair_1 = require("../models/LabelsBetweenObjectPair");
const ModelError_1 = require("../models/ModelError");
const MultiAssociatedObjectWithLabel_1 = require("../models/MultiAssociatedObjectWithLabel");
const NextPage_1 = require("../models/NextPage");
const Paging_1 = require("../models/Paging");
const PreviousPage_1 = require("../models/PreviousPage");
const PublicAssociationDefinitionCreateRequest_1 = require("../models/PublicAssociationDefinitionCreateRequest");
const PublicAssociationDefinitionUpdateRequest_1 = require("../models/PublicAssociationDefinitionUpdateRequest");
const PublicAssociationMultiArchive_1 = require("../models/PublicAssociationMultiArchive");
const PublicAssociationMultiPost_1 = require("../models/PublicAssociationMultiPost");
const PublicAssociationMultiWithLabel_1 = require("../models/PublicAssociationMultiWithLabel");
const PublicDefaultAssociation_1 = require("../models/PublicDefaultAssociation");
const PublicDefaultAssociationMultiPost_1 = require("../models/PublicDefaultAssociationMultiPost");
const PublicFetchAssociationsBatchRequest_1 = require("../models/PublicFetchAssociationsBatchRequest");
const PublicObjectId_1 = require("../models/PublicObjectId");
const StandardError_1 = require("../models/StandardError");
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
const supportedMediaTypes = {
    "application/json": Infinity,
    "application/octet-stream": 0,
    "application/x-www-form-urlencoded": 0
};
let enumsMap = new Set([
    "AssociationSpecAssociationCategoryEnum",
    "AssociationSpecWithLabelCategoryEnum",
    "BatchResponseLabelsBetweenObjectPairStatusEnum",
    "BatchResponseLabelsBetweenObjectPairWithErrorsStatusEnum",
    "BatchResponsePublicAssociationMultiWithLabelStatusEnum",
    "BatchResponsePublicAssociationMultiWithLabelWithErrorsStatusEnum",
    "BatchResponsePublicDefaultAssociationStatusEnum",
    "ErrorCategoryHttpStatusEnum",
]);
let typeMap = {
    "AssociationSpec": AssociationSpec_1.AssociationSpec,
    "AssociationSpecWithLabel": AssociationSpecWithLabel_1.AssociationSpecWithLabel,
    "BatchInputPublicAssociationMultiArchive": BatchInputPublicAssociationMultiArchive_1.BatchInputPublicAssociationMultiArchive,
    "BatchInputPublicAssociationMultiPost": BatchInputPublicAssociationMultiPost_1.BatchInputPublicAssociationMultiPost,
    "BatchInputPublicDefaultAssociationMultiPost": BatchInputPublicDefaultAssociationMultiPost_1.BatchInputPublicDefaultAssociationMultiPost,
    "BatchInputPublicFetchAssociationsBatchRequest": BatchInputPublicFetchAssociationsBatchRequest_1.BatchInputPublicFetchAssociationsBatchRequest,
    "BatchResponseLabelsBetweenObjectPair": BatchResponseLabelsBetweenObjectPair_1.BatchResponseLabelsBetweenObjectPair,
    "BatchResponseLabelsBetweenObjectPairWithErrors": BatchResponseLabelsBetweenObjectPairWithErrors_1.BatchResponseLabelsBetweenObjectPairWithErrors,
    "BatchResponsePublicAssociationMultiWithLabel": BatchResponsePublicAssociationMultiWithLabel_1.BatchResponsePublicAssociationMultiWithLabel,
    "BatchResponsePublicAssociationMultiWithLabelWithErrors": BatchResponsePublicAssociationMultiWithLabelWithErrors_1.BatchResponsePublicAssociationMultiWithLabelWithErrors,
    "BatchResponsePublicDefaultAssociation": BatchResponsePublicDefaultAssociation_1.BatchResponsePublicDefaultAssociation,
    "CollectionResponseAssociationSpecWithLabelNoPaging": CollectionResponseAssociationSpecWithLabelNoPaging_1.CollectionResponseAssociationSpecWithLabelNoPaging,
    "ErrorCategory": ErrorCategory_1.ErrorCategory,
    "ErrorDetail": ErrorDetail_1.ErrorDetail,
    "LabelsBetweenObjectPair": LabelsBetweenObjectPair_1.LabelsBetweenObjectPair,
    "ModelError": ModelError_1.ModelError,
    "MultiAssociatedObjectWithLabel": MultiAssociatedObjectWithLabel_1.MultiAssociatedObjectWithLabel,
    "NextPage": NextPage_1.NextPage,
    "Paging": Paging_1.Paging,
    "PreviousPage": PreviousPage_1.PreviousPage,
    "PublicAssociationDefinitionCreateRequest": PublicAssociationDefinitionCreateRequest_1.PublicAssociationDefinitionCreateRequest,
    "PublicAssociationDefinitionUpdateRequest": PublicAssociationDefinitionUpdateRequest_1.PublicAssociationDefinitionUpdateRequest,
    "PublicAssociationMultiArchive": PublicAssociationMultiArchive_1.PublicAssociationMultiArchive,
    "PublicAssociationMultiPost": PublicAssociationMultiPost_1.PublicAssociationMultiPost,
    "PublicAssociationMultiWithLabel": PublicAssociationMultiWithLabel_1.PublicAssociationMultiWithLabel,
    "PublicDefaultAssociation": PublicDefaultAssociation_1.PublicDefaultAssociation,
    "PublicDefaultAssociationMultiPost": PublicDefaultAssociationMultiPost_1.PublicDefaultAssociationMultiPost,
    "PublicFetchAssociationsBatchRequest": PublicFetchAssociationsBatchRequest_1.PublicFetchAssociationsBatchRequest,
    "PublicObjectId": PublicObjectId_1.PublicObjectId,
    "StandardError": StandardError_1.StandardError,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap.has(expectedType)) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType;
            }
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType;
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType;
                    }
                    else {
                        return expectedType;
                    }
                }
                else {
                    return expectedType;
                }
            }
        }
    }
    static serialize(data, type, format) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) {
            let subType = type.replace("Array<", "");
            subType = subType.substring(0, subType.length - 1);
            let transformedData = [];
            for (let date of data) {
                transformedData.push(ObjectSerializer.serialize(date, subType, format));
            }
            return transformedData;
        }
        else if (type === "Date") {
            if (format == "date") {
                let month = data.getMonth() + 1;
                month = month < 10 ? "0" + month.toString() : month.toString();
                let day = data.getDate();
                day = day < 10 ? "0" + day.toString() : day.toString();
                return data.getFullYear() + "-" + month + "-" + day;
            }
            else {
                return data.toISOString();
            }
        }
        else {
            if (enumsMap.has(type)) {
                return data;
            }
            if (!typeMap[type]) {
                return data;
            }
            type = this.findCorrectType(data, type);
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let attributeType of attributeTypes) {
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type, attributeType.format);
            }
            return instance;
        }
    }
    static deserialize(data, type, format) {
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) {
            let subType = type.replace("Array<", "");
            subType = subType.substring(0, subType.length - 1);
            let transformedData = [];
            for (let date of data) {
                transformedData.push(ObjectSerializer.deserialize(date, subType, format));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap.has(type)) {
                return data;
            }
            if (!typeMap[type]) {
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let attributeType of attributeTypes) {
                let value = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type, attributeType.format);
                if (value !== undefined) {
                    instance[attributeType.name] = value;
                }
            }
            return instance;
        }
    }
    static normalizeMediaType(mediaType) {
        if (mediaType === undefined) {
            return undefined;
        }
        return mediaType.split(";")[0].trim().toLowerCase();
    }
    static getPreferredMediaType(mediaTypes) {
        if (!mediaTypes) {
            return "application/json";
        }
        const normalMediaTypes = mediaTypes.map(this.normalizeMediaType);
        let selectedMediaType = undefined;
        let selectedRank = -Infinity;
        for (const mediaType of normalMediaTypes) {
            if (supportedMediaTypes[mediaType] > selectedRank) {
                selectedMediaType = mediaType;
                selectedRank = supportedMediaTypes[mediaType];
            }
        }
        if (selectedMediaType === undefined) {
            throw new Error("None of the given media types are supported: " + mediaTypes.join(", "));
        }
        return selectedMediaType;
    }
    static stringify(data, mediaType) {
        if (mediaType === "text/plain") {
            return String(data);
        }
        if (mediaType === "application/json") {
            return JSON.stringify(data);
        }
        throw new Error("The mediaType " + mediaType + " is not supported by ObjectSerializer.stringify.");
    }
    static parse(rawData, mediaType) {
        if (mediaType === undefined) {
            throw new Error("Cannot parse content. No Content-Type defined.");
        }
        if (mediaType === "text/plain") {
            return rawData;
        }
        if (mediaType === "application/json") {
            return JSON.parse(rawData);
        }
        if (mediaType === "text/html") {
            return rawData;
        }
        throw new Error("The mediaType " + mediaType + " is not supported by ObjectSerializer.parse.");
    }
}
exports.ObjectSerializer = ObjectSerializer;
