// @hubspot/api-client/lib/src/discovery/cms/source_code/SourceCodeDiscovery.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const index_1 = require("../../../../codegen/cms/source_code/index");
const rxjsStub_1 = require("../../../../codegen/cms/source_code/rxjsStub");
const ApiClientConfigurator_1 = require("../../../configuration/ApiClientConfigurator");
const ApiDecoratorService_1 = __importDefault(require("../../../services/ApiDecoratorService"));
class SourceCodeDiscovery {
    constructor(config) {
        const configuration = (0, index_1.createConfiguration)(ApiClientConfigurator_1.ApiClientConfigurator.getParams(config, index_1.ServerConfiguration, rxjsStub_1.Observable, rxjsStub_1.Observable));
        this.contentApi = ApiDecoratorService_1.default.getInstance().apply(new index_1.ContentApi(configuration));
        this.extractApi = ApiDecoratorService_1.default.getInstance().apply(new index_1.ExtractApi(configuration));
        this.metadataApi = ApiDecoratorService_1.default.getInstance().apply(new index_1.MetadataApi(configuration));
        this.sourceCodeExtractApi = ApiDecoratorService_1.default.getInstance().apply(new index_1.SourceCodeExtractApi(configuration));
        this.validationApi = ApiDecoratorService_1.default.getInstance().apply(new index_1.ValidationApi(configuration));
    }
}
exports.default = SourceCodeDiscovery;
