// @pandium/apollo-io-client/dist/schemas/updateAccountOwnershipSchema.js
import { z } from "zod";
/**
 * @description 401
 */
export var updateAccountOwnership401Schema = z.any();
/**
      * @description 403
      */
export var updateAccountOwnership403Schema = z.object({ "error": z.string().nullish(), "error_code": z.string().nullish() });
/**
      * @description 422
      */
export var updateAccountOwnership422Schema = z.union([z.object({ "error": z.string().nullish() }), z.object({ "error": z.string().nullish() })]);
/**
      * @description 429
      */
export var updateAccountOwnership429Schema = z.object({ "message": z.string().nullish() });
/**
      * @description 200
      */
export var updateAccountOwnershipMutationResponseSchema = z.object({ "accounts": z.array(z.object({ "id": z.string().nullish(), "owner_id": z.string().nullish(), "crm_owner_id": z.any().nullish() })).nullish() });
export var updateAccountOwnershipQueryParamsSchema = z.object({ "account_ids[]": z.array(z.string()).describe("The Apollo IDs for the account that you want to assign to an owner. <br><br>To find account IDs, call the <a href=\"https://docs.apollo.io/reference/search-for-accounts\" target=\"_blank\">Search for Accounts endpoint</a> and identify the 'id' value for the account. <br><br>Example: '66e9abf95ac32901b20d1a0d'"), "owner_id": z.string().describe("The ID for the account owner within your team's Apollo account. This user will be assigned ownership of the accounts. <br><br>Use the <a href=\"https://docs.apollo.io/reference/get-a-list-of-users\" target=\"_blank\">Get a List of Users endpoint</a> to retrieve IDs for all of the users within your Apollo account. <br><br>Example: '66302798d03b9601c7934ebf'") });
