// @hubspot/api-client/lib/src/enums/AssociationTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssociationTypes = void 0;
var AssociationTypes;
(function (AssociationTypes) {
    AssociationTypes[AssociationTypes["primaryContactToCompany"] = 1] = "primaryContactToCompany";
    AssociationTypes[AssociationTypes["primaryCompanyToContact"] = 2] = "primaryCompanyToContact";
    AssociationTypes[AssociationTypes["dealToContact"] = 3] = "dealToContact";
    AssociationTypes[AssociationTypes["contactToDeal"] = 4] = "contactToDeal";
    AssociationTypes[AssociationTypes["primaryDealToCompany"] = 5] = "primaryDealToCompany";
    AssociationTypes[AssociationTypes["primaryCompanyToDeal"] = 6] = "primaryCompanyToDeal";
    AssociationTypes[AssociationTypes["contactToTicket"] = 15] = "contactToTicket";
    AssociationTypes[AssociationTypes["ticketToContact"] = 16] = "ticketToContact";
    AssociationTypes[AssociationTypes["dealToLineItem"] = 19] = "dealToLineItem";
    AssociationTypes[AssociationTypes["lineItemToDeal"] = 20] = "lineItemToDeal";
    AssociationTypes[AssociationTypes["primaryCompanyToTicket"] = 25] = "primaryCompanyToTicket";
    AssociationTypes[AssociationTypes["primaryTicketToCompany"] = 26] = "primaryTicketToCompany";
    AssociationTypes[AssociationTypes["dealToTicket"] = 27] = "dealToTicket";
    AssociationTypes[AssociationTypes["ticketToDeal"] = 28] = "ticketToDeal";
    AssociationTypes[AssociationTypes["dealToQuote"] = 63] = "dealToQuote";
    AssociationTypes[AssociationTypes["quoteToDeal"] = 64] = "quoteToDeal";
    AssociationTypes[AssociationTypes["quoteToLineItem"] = 67] = "quoteToLineItem";
    AssociationTypes[AssociationTypes["lineItemToQuote"] = 68] = "lineItemToQuote";
    AssociationTypes[AssociationTypes["quoteToContact"] = 69] = "quoteToContact";
    AssociationTypes[AssociationTypes["contactToQuote"] = 70] = "contactToQuote";
    AssociationTypes[AssociationTypes["quoteToCompany"] = 71] = "quoteToCompany";
    AssociationTypes[AssociationTypes["companyToQuote"] = 72] = "companyToQuote";
    AssociationTypes[AssociationTypes["contactToFeedbackSubmission"] = 97] = "contactToFeedbackSubmission";
    AssociationTypes[AssociationTypes["feedbackSubmissionToContact"] = 98] = "feedbackSubmissionToContact";
    AssociationTypes[AssociationTypes["ticketToFeedbackSubmission"] = 99] = "ticketToFeedbackSubmission";
    AssociationTypes[AssociationTypes["feedbackSubmissionToTicket"] = 100] = "feedbackSubmissionToTicket";
    AssociationTypes[AssociationTypes["companyToCall"] = 181] = "companyToCall";
    AssociationTypes[AssociationTypes["callToCompany"] = 182] = "callToCompany";
    AssociationTypes[AssociationTypes["companyToEmail"] = 185] = "companyToEmail";
    AssociationTypes[AssociationTypes["emailToCompany"] = 186] = "emailToCompany";
    AssociationTypes[AssociationTypes["companyToMeeting"] = 187] = "companyToMeeting";
    AssociationTypes[AssociationTypes["meetingToCompany"] = 188] = "meetingToCompany";
    AssociationTypes[AssociationTypes["companyToNote"] = 189] = "companyToNote";
    AssociationTypes[AssociationTypes["noteToCompany"] = 190] = "noteToCompany";
    AssociationTypes[AssociationTypes["companyToTask"] = 191] = "companyToTask";
    AssociationTypes[AssociationTypes["taskToCompany"] = 192] = "taskToCompany";
    AssociationTypes[AssociationTypes["contactToCall"] = 193] = "contactToCall";
    AssociationTypes[AssociationTypes["callToContact"] = 194] = "callToContact";
    AssociationTypes[AssociationTypes["contactToEmail"] = 197] = "contactToEmail";
    AssociationTypes[AssociationTypes["emailToContact"] = 198] = "emailToContact";
    AssociationTypes[AssociationTypes["contactToMeeting"] = 199] = "contactToMeeting";
    AssociationTypes[AssociationTypes["meetingToContact"] = 200] = "meetingToContact";
    AssociationTypes[AssociationTypes["contactToNote"] = 201] = "contactToNote";
    AssociationTypes[AssociationTypes["noteToContact"] = 202] = "noteToContact";
    AssociationTypes[AssociationTypes["contactToTask"] = 203] = "contactToTask";
    AssociationTypes[AssociationTypes["taskToContact"] = 204] = "taskToContact";
    AssociationTypes[AssociationTypes["dealToCall"] = 205] = "dealToCall";
    AssociationTypes[AssociationTypes["callToDeal"] = 206] = "callToDeal";
    AssociationTypes[AssociationTypes["dealToEmail"] = 209] = "dealToEmail";
    AssociationTypes[AssociationTypes["emailToDeal"] = 210] = "emailToDeal";
    AssociationTypes[AssociationTypes["dealToMeeting"] = 211] = "dealToMeeting";
    AssociationTypes[AssociationTypes["meetingToDeal"] = 212] = "meetingToDeal";
    AssociationTypes[AssociationTypes["dealToNote"] = 213] = "dealToNote";
    AssociationTypes[AssociationTypes["noteToDeal"] = 214] = "noteToDeal";
    AssociationTypes[AssociationTypes["dealToTask"] = 215] = "dealToTask";
    AssociationTypes[AssociationTypes["taskToDeal"] = 216] = "taskToDeal";
    AssociationTypes[AssociationTypes["quoteToTask"] = 217] = "quoteToTask";
    AssociationTypes[AssociationTypes["taskToQuote"] = 218] = "taskToQuote";
    AssociationTypes[AssociationTypes["ticketToCall"] = 219] = "ticketToCall";
    AssociationTypes[AssociationTypes["callToTicket"] = 220] = "callToTicket";
    AssociationTypes[AssociationTypes["ticketToEmail"] = 223] = "ticketToEmail";
    AssociationTypes[AssociationTypes["emailToTicket"] = 224] = "emailToTicket";
    AssociationTypes[AssociationTypes["ticketToMeeting"] = 225] = "ticketToMeeting";
    AssociationTypes[AssociationTypes["meetingToTicket"] = 226] = "meetingToTicket";
    AssociationTypes[AssociationTypes["ticketToNote"] = 227] = "ticketToNote";
    AssociationTypes[AssociationTypes["noteToTicket"] = 228] = "noteToTicket";
    AssociationTypes[AssociationTypes["ticketToTask"] = 229] = "ticketToTask";
    AssociationTypes[AssociationTypes["taskToTicket"] = 230] = "taskToTicket";
    AssociationTypes[AssociationTypes["contactToCompany"] = 279] = "contactToCompany";
    AssociationTypes[AssociationTypes["companyToContact"] = 280] = "companyToContact";
    AssociationTypes[AssociationTypes["ticketToCompany"] = 339] = "ticketToCompany";
    AssociationTypes[AssociationTypes["companyToTicket"] = 340] = "companyToTicket";
    AssociationTypes[AssociationTypes["dealToCompany"] = 341] = "dealToCompany";
    AssociationTypes[AssociationTypes["companyToDeal"] = 342] = "companyToDeal";
    AssociationTypes[AssociationTypes["callToMeeting"] = 399] = "callToMeeting";
    AssociationTypes[AssociationTypes["meetingToCall"] = 400] = "meetingToCall";
    AssociationTypes[AssociationTypes["postalMailToContact"] = 453] = "postalMailToContact";
    AssociationTypes[AssociationTypes["contactToPostalMail"] = 454] = "contactToPostalMail";
    AssociationTypes[AssociationTypes["postalMailToTicket"] = 455] = "postalMailToTicket";
    AssociationTypes[AssociationTypes["ticketToPostalMail"] = 456] = "ticketToPostalMail";
    AssociationTypes[AssociationTypes["postalMailToDeal"] = 457] = "postalMailToDeal";
    AssociationTypes[AssociationTypes["dealToPostalMail"] = 458] = "dealToPostalMail";
    AssociationTypes[AssociationTypes["postalMailToCompany"] = 459] = "postalMailToCompany";
    AssociationTypes[AssociationTypes["companyToPostalMail"] = 460] = "companyToPostalMail";
})(AssociationTypes = exports.AssociationTypes || (exports.AssociationTypes = {}));
